---
title: "RNA viruses"
author: "Ella Sieradzki"
date: "2022-12-27"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
library("tidyverse")
library("data.table")
library("ggplot2")
library("vegan")
library("RColorBrewer")
library("ape")
library("ggpubr")
library("ggpmisc")
library("microshades")
library("DESeq2")
library("EnhancedVolcano")
library("pheatmap")
library("dichromat")
library("stringi")
library("dunn.test")
library("chemodiv")
```

## Load data

```{r votu table, echo=FALSE}
votu <- read.delim("coverm_tpm_breadth50_mmseqs_Gary.tsv", sep="\t", header=T, stringsAsFactors = F)
str(votu)
#colnames(votu) <- gsub("_sort\\.RPKM", "", colnames(votu))
colnames(votu) <- gsub("_sort\\.TPM", "", colnames(votu))
votus_rmv <- scan(file = "vOTUs_to_remove_unclassified.txt",what = character())
votu <- votu %>%
  filter(!(Contig %in% votus_rmv))

wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]
# To account for T0 under both treatments (P amended and unamended), replicate the T0 values and assign once to P and once to noP
# toadd <- wetup[, 1:3]
# colnames(toadd) <- gsub("A$", "P.A", colnames(toadd))
# colnames(toadd) <- gsub("B$", "P.B", colnames(toadd))
# colnames(toadd) <- gsub("C$", "P.C", colnames(toadd))
# wetup <- data.frame(cbind(toadd, wetup))

wetup <- t(wetup)

design <- read.delim("mtx_design_Gary.txt")
#design <- read.delim("mtx_design_Gary_2T0.txt")
```

## Create Permanova

```{r}
adonis2(wetup~time*Phos*isotope, data=design)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = wetup ~ time * Phos * isotope, data = design)
#                   Df SumOfSqs      R2      F Pr(>F)    
# time               1   0.5255 0.05696 2.8602  0.002 ** 
# Phos               1   1.3685 0.14834 7.4487  0.001 ***
# isotope            1   0.5500 0.05962 2.9936  0.002 ** 
# time:Phos          1   0.3647 0.03953 1.9848  0.014 *  
# time:isotope       1   0.2528 0.02741 1.3762  0.110    
# Phos:isotope       1   0.2411 0.02614 1.3125  0.150    
# time:Phos:isotope  1   0.2273 0.02464 1.2372  0.187    
# Residual          31   5.6956 0.61737                  
# Total             38   9.2257 1.00000                  
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

# PERMANOVA without T0
design_no_T0 <- design %>% filter(time>0)
wetup_no_T0 <- wetup[-(1:3),]
adonis2(wetup_no_T0~time*Phos*isotope, data=design_no_T0)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = wetup_no_T0 ~ time * Phos * isotope, data = design_no_T0)
#                   Df SumOfSqs      R2      F Pr(>F)    
# time               1   0.3802 0.04664 2.1474  0.009 ** 
# Phos               1   1.2943 0.15879 7.3108  0.001 ***
# isotope            1   0.5977 0.07333 3.3760  0.001 ***
# time:Phos          1   0.2715 0.03331 1.5337  0.078 .  
# time:isotope       1   0.2265 0.02779 1.2796  0.156    
# Phos:isotope       1   0.2533 0.03108 1.4309  0.110    
# time:Phos:isotope  1   0.1702 0.02088 0.9613  0.434    
# Residual          28   4.9569 0.60817                  
# Total             35   8.1506 1.00000                  
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

# Square root transformation
wetup_sqrt <- sqrt(wetup)
wetup_dist <- vegdist(wetup_sqrt, method="bray", na.rm = T)
wetup_cols <- factor(design$cosm)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_")) %>%
  mutate(ellps = as.factor(gsub("T[1-3]_", "", cond)))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample, group=ellps)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  #stat_ellipse() +
  theme_bw()
ggsave("Gary_RNA_viruses_PCoA_wetup_sqrt_tpm.pdf", width=8)
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample, group=ellps)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  theme_bw()
ggsave("Gary_RNA_viruses_PCoA_wetup_sqrt_tpm_noellipses.pdf", width=10)
# Recalculate without T0 to see if later temporal patterns pop up
wetup_sqrt <- wetup_sqrt[!(rownames(wetup_sqrt) %like% "T0"),]
wetup_dist <- vegdist(wetup_sqrt, method="bray", na.rm = T)
pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  #geom_text(size=1) +
  theme_bw()
# Nah, no epiphany here 
```

```{r}
# vOTU sharing between replicates, supplemental figure
wetup_df <- votu %>% column_to_rownames("Contig")
wetup_df <- wetup_df[rowSums(wetup_df) > 0,]
wetup_df[wetup_df > 0] <- 1 # convert coverage to presence/absence

sumsrel <- c()
sumsPrel <- c()

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "Dry.soil.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsrel <- c(sumsrel, sums2/sum(sums2))
fig0 <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H218O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsPrel <- c(sumsrel, sums2/sum(sums2))
figT1_18O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H218O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsrel <- c(sumsrel, sums2/sum(sums2))
figT1_18O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H2O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsPrel <- c(sumsrel, sums2/sum(sums2))
figT1_16O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H2O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsrel <- c(sumsrel, sums2/sum(sums2))
figT1_16O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H218O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsPrel <- c(sumsrel, sums2/sum(sums2))
figT2_18O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H218O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsrel <- c(sumsrel, sums2/sum(sums2))
figT2_18O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H2O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsPrel <- c(sumsrel, sums2/sum(sums2))
figT2_16O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H2O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsrel <- c(sumsrel, sums2/sum(sums2))
figT2_16O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H218O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT3_18O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H218O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsrel <- c(sumsrel, sums2/sum(sums2))
figT3_18O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H2O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsPrel <- c(sumsrel, sums2/sum(sums2))
figT3_16O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H2O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
sumsrel <- c(sumsrel, sums2/sum(sums2))
figT3_16O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

ggarrange(fig0, 
          ggarrange(figT1_18O_P, figT1_18O_noP, figT1_16O_P, figT1_16O_noP, nrow=1, ncol=4, common.legend = T),
          ggarrange(figT2_18O_P, figT2_18O_noP, figT2_16O_P, figT2_16O_noP, nrow=1, ncol=4, common.legend = T),
          ggarrange(figT3_18O_P, figT3_18O_noP, figT3_16O_P, figT3_16O_noP, nrow=1, ncol=4, common.legend = T),
          nrow = 4)
ggsave("fig_S1_replication_hist_tpm.pdf", width=8, height=10)

# As a boxplot over all time points with/without P
sumsrel_df <- data.frame(cbind(sumsrel[c(1,4,7,10,13,16,19)], sumsrel[c(1,4,7,10,13,16,19)+1], sumsrel[c(1,4,7,10,13,16,19)+2])) %>% pivot_longer(cols = everything(), names_to = "repNum")
sumsPrel_df <- data.frame(cbind(sumsPrel[c(1,4,7,10,13,16,19)], sumsPrel[c(1,4,7,10,13,16,19)+1], sumsPrel[c(1,4,7,10,13,16,19)+2])) %>% pivot_longer(cols = everything(), names_to = "repNum")
p1 <- ggplot(data = sumsrel_df, aes(x = repNum, y = value)) + geom_boxplot(aes(fill = repNum)) + theme_classic() + scale_fill_brewer(palette = "Purples") + 
  xlab("Number of replicates") + ylab("Proportion of vOTUs present in sample") + ggtitle("Unamended soil")
p2 <- ggplot(data = sumsPrel_df, aes(x = repNum, y = value)) + geom_boxplot(aes(fill = repNum)) + theme_classic() + scale_fill_brewer(palette = "Purples") + 
  xlab("Number of replicates") + ylab("Proportion of vOTUs present in sample") + ggtitle("Phosphate amended soil")
ggarrange(p1, p2, common.legend = T, nrow = 1, ncol = 2)
ggsave("sup_fig_S4_replication.pdf", width = 7)

# Number of samples each vOTU appears in histogram
wetup_df <- wetup_df %>% select(-matches("T0.*P.*"))
sums <- rowSums(wetup_df)

hist(sums, breaks=39, xlab = "Number of microcosms", ylab = "Number of vOTUs", main = "")
```

```{r}
# Taxonomy distribution
wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]

tax <- read.delim("vOTU_final_taxonomy_by_tree.txt", sep="\t", header=F, stringsAsFactors = F)
colnames(tax) <- c("Contig", "taxonomy")
ictv <- read.delim("ICTV_RNA_viruses.txt", sep="\t", header=T, stringsAsFactors = F)
tax$Class <- tax$taxonomy

getOrder <- function(taxname) {
  if ((taxname %like% "ota$") | (taxname %like% "tes$") | (taxname %like% "les$") | (taxname %like% "Unassigned")) return(taxname) # if it's already a phylum or class name keep it as is
  if (taxname %in% c("Zhaovirus", "Yanvirus", "Permutotetraviridae", "Ophioviridae")) return("Unassigned") # These exceptions don't have a class or phylum
  if (taxname %like% "dae$") return(unique(ictv$Order[ictv$Family == taxname]))
  if (taxname %in% c("o.0012", "o.0014", "o.0028")) return("Pisuviricota") # New clades from Neri 2022
  if (taxname %in% c("o.0063","o.0067")) return("Negarnaviricota") # New clades from Neri 2022
  if (taxname %in% c("o.0036", "o.0042", "o.0050", "o.0051", "o.0435")) return("Kitrinoviricota") # New clades from Neri 2022
  if (taxname == "o.0081") return("Lenarviricota") # New clades from Neri 2022
  else return(unique(ictv$Order[ictv$Genus == taxname]))
}
getClass <- function(taxname) {
  if ((taxname %like% "ota$") | (taxname %like% "tes$") | (taxname %like% "Unassigned")) return(taxname) # if it's already a phylum or class name keep it as is
  if (taxname == "Levivirales") return("Leviviricetes") # Levivirales don't exist anymore so I had to define this manually
  if (taxname == "Ophioviridae") return("Milneviricetes") # Ophioviridae is a new family so I had to define this manually
  if (taxname %in% c("Zhaovirus", "Yanvirus", "Permutotetraviridae")) return("Unassigned") # These exceptions don't have a class or phylum
  if (taxname %in% c("o.0012", "o.0014", "o.0028")) return("Pisuviricota") # New clades from Neri 2022
  if (taxname %in% c("o.0063","o.0067")) return("Negarnaviricota") # New clades from Neri 2022
  if (taxname %in% c("o.0036", "o.0042", "o.0050", "o.0051", "o.0435")) return("Kitrinoviricota") # New clades from Neri 2022
  if (taxname == "o.0081") return("Lenarviricota") # New clades from Neri 2022
  if (taxname %like% "les$") return(unique(ictv$Class[ictv$Order == taxname]))
  if (taxname %like% "dae$") return(unique(ictv$Class[ictv$Family == taxname]))
  else return(unique(ictv$Class[ictv$Genus == taxname]))
}

tax$Class <- unlist(sapply(tax$taxonomy, getClass))
tax$Order <- unlist(sapply(tax$taxonomy, getOrder))

write.csv(tax, "Order_level_tax.tsv", row.names = FALSE)
# To load the existing table
tax <- read.csv("Order_level_tax.tsv")

wetup_tax <- wetup %>% rownames_to_column("Contig") %>% left_join(tax, by="Contig")

wetup_cols <- which(colnames(wetup_tax) %like% "^T")
wetup <- wetup_tax %>% select(Contig, taxonomy, starts_with("T")) %>%
  filter(rowSums(wetup_tax[,wetup_cols]) > 0, na.rm = TRUE) %>% unique() # 2190 vOTUs

wetup_pvt <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Class, Order, taxonomy), names_to="variable", values_to="value") %>%
  mutate(pvt = ifelse(value > 0, 1, 0)) %>%
  group_by(Class, variable) %>%
  summarize_at("pvt", sum) %>%
  mutate(variable = as.character(variable))
top11 <- wetup_pvt %>% group_by(Class) %>% summarize_at("pvt", sum)
top11 <- head(top11$Class[order(-top11$pvt)], 10)
wetup_tax_plot <- wetup_pvt %>%
  mutate(Class = ifelse(Class %in% top11, Class, "Other")) %>%
  mutate(Class = factor(Class))
top11 <- c(top11, "Other")
wetup_tax_plot$Class <- fct_relevel(wetup_tax_plot$Class, top11)
# Bar chart
ggplot(data = wetup_tax_plot, aes(x = variable, y = pvt, fill=Class)) + 
  geom_bar(stat = "identity", position = "stack") + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("Sample") + ylab("Number of vOTUs present")
ggsave("wetup_richness_tax_bar_class.pdf")
# Boxplot
P1 <- ggplot(data = wetup_tax_plot, aes(x = Class, y = pvt, fill=Class)) + 
  geom_boxplot() + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("taxonomy") + ylab("Number of vOTUs per sample")
ggsave(P1, "wetup_richness_tax_box_class.pdf")
# Relative abundance stacked barchart
for_durna <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Class, Order, taxonomy), names_to="variable", values_to="tpm") %>%
  group_by(Order, variable) %>%
  summarize_at("tpm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = tpm*100/sum(tpm))
wetup_pct <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Class, Order, taxonomy), names_to="variable", values_to="tpm") %>%
  group_by(Class, variable) %>%
  summarize_at("tpm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = tpm*100/sum(tpm))

wetup_tax_plot <- wetup_pct %>% 
  mutate(Class = ifelse(Class %in% top11, Class, "Other")) %>%
  mutate(Class = factor(Class))
wetup_tax_plot$Class <- fct_relevel(wetup_tax_plot$Class, top11)

# Most phages are Leviviricetes or Durnavirales
levi <- wetup_tax_plot %>% filter(Class %like% "Levivi") # Save Leviviricetes for later script
durna <- for_durna %>% filter(Order %like% "Durnavi") # Save Durnavirales phages for later (see Neri 2022)

ggplot(data = wetup_tax_plot, aes(x = variable, y = pct, fill=Class)) + 
  geom_bar(stat = "identity", position = "fill") + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("Sample") + ylab("Relative abundance")
ggsave("wetup_relabun_tax_bar_tpm_class.pdf")
# Boxplot
P2 <- ggplot(data = wetup_tax_plot, aes(x = Class, y = pct, fill=Class)) + 
  geom_boxplot() + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("taxonomy") + ylab("Relative abundance per sample")
ggsave(P2, "wetup_relabun_tax_box_tpm_class.pdf")

# Is there a correlation between richness and abundance?
temp <- wetup_pct %>% left_join(wetup_pvt, by=c("Class", "variable"))
temp$Class[!(temp$Class %in% top11)] <- "Other"
temp$Class <- fct_relevel(temp$Class, top11)
P3 <- ggplot(data=temp, aes(x=pvt, y=pct)) + geom_point(aes(fill=Class), colour="black",pch=21, size=2) + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + theme(legend.position="none") + 
  stat_poly_line(color="black") +
  stat_poly_eq(use_label(c("eq", "adj.R2", "f", "p", "n"))) +
  xlab("Number of unique vOTUs from taxonomic group")+ ylab("Relative abundance of vOTUs from taxonomic group")
ggsave("wetup_richness_vs_abundance_class.pdf")
```

```{r}
# Special vOTUs
dry_cols <- which(colnames(wetup_tax) %like% "T0")
p_cols <- which(colnames(wetup_tax) %like% "P")
nop_cols <- c(which(colnames(wetup_tax) %like% "T[1-3].*O.[A-C]"))

# Only in dry soil
tmp1 <- wetup_tax[rowSums(wetup_tax[,dry_cols]) > 0 & rowSums(wetup_tax[,c(p_cols, nop_cols)]) == 0,]
nrow(tmp1) #31
# Only in P amended soil
tmp2 <- wetup_tax[rowSums(wetup_tax[,p_cols]) > 0 & rowSums(wetup_tax[,nop_cols]) == 0,]
nrow(tmp2) #327
# Only in unamended soil
tmp3 <- wetup_tax[rowSums(wetup_tax[,p_cols]) == 0 & rowSums(wetup_tax[,nop_cols]) > 0,]
nrow(tmp3) #438

# Unify legend
cls <- unique(c(tmp1$Class, tmp2$Class, tmp3$Class))
phl <- read.delim("cls_to_phl.txt", sep = "\t", header = F)
colnames(phl) <- c("Class", "Phylum")
pal <- phl %>% group_by(Phylum) %>% tally()
pal$palette <- c("micro_purple", "micro_orange", "micro_blue", "micro_brown", "micro_green", "micro_gray")
cols <- c()
for (i in 1:nrow(pal)) {
  cols <- c(cols, microshades_palette(pal$palette[i], pal$n[i]))
}
phl <- phl[order(phl$Phylum),]
phl$cols <- cols
names(cols) <- phl$Class

tmp_long <- tmp1 %>% select(starts_with("T0"), "Order", "Class") %>% pivot_longer(cols = starts_with("T0"), names_to = "sample") %>% group_by(Class) %>% summarise(total = sum(value)) %>% arrange(desc(total))
# tmp_long$Class <- fct_relevel(tmp_long$Class, tmp_long$Class)
# p1 <- ggplot(tmp_long, aes(x = 1, y=total)) + geom_bar(stat = "identity", aes(fill = Class), position = "stack") + theme_bw() + scale_fill_manual(values = cols) + xlab("Dry soil, N=42") + ylab("Total TPM")

tmp_long2 <- tmp2 %>% select(contains("P"), "Order", "Class") %>% pivot_longer(cols = contains("P"), names_to = "sample") %>% group_by(Class) %>% summarise(total = sum(value)) %>% arrange(desc(total))
# top11 <- tmp_long$Class[1:11]
# tmp_long$Class[!(tmp_long$Class %in% top11)] <- "Other"
# tmp_long$Class <- fct_relevel(tmp_long$Class, c(top11, "Other"))
# p2 <- ggplot(tmp_long, aes(x = 1, y=total)) + geom_bar(stat = "identity", aes(fill = Class), position = "stack") + theme_bw() + scale_fill_manual(values = cols) + xlab("P amended soil, N=525") + ylab("Total TPM")

tmp_long3 <- tmp3 %>% select(matches("T[1-3].*O.[A-C]"), "Order", "Class") %>% pivot_longer(cols =starts_with("T"), names_to = "sample") %>% group_by(Class) %>% summarise(total = sum(value)) %>% arrange(desc(total))

tmp_long_total <- tmp_long %>% full_join(tmp_long2, by = "Class") %>% full_join(tmp_long3, by = "Class")
colnames(tmp_long_total) <- c("Class", "DrySoil", "Phosphate", "Unamended")
tmp_long_total <- tmp_long_total %>% pivot_longer(cols = !Class, names_to = "SampleType", values_to = "TPM") %>% replace(is.na(.), 0)

ggplot(tmp_long_total, aes(x = SampleType, y=TPM)) + geom_bar(stat = "identity", aes(fill = Class)) + theme_bw() + scale_fill_manual(values = cols) + xlab("Sample Type") + ylab("Total TPM")
ggsave("sup_fig_S5_special_vOTUs.pdf", width = 7, height = 8)

# ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
# ggsave("fig_S-onlyin.pdf", width = 8.5)

# vOTUs fueled by P
tmp2_mat <- tmp2 %>% select("Contig", starts_with("T", ignore.case = FALSE)) %>% pivot_longer(cols = !Contig, names_to = "sample", values_to = "TPM") %>% mutate(time = substr(sample, 1, 2)) %>% mutate(amnd = str_detect(sample, "P")) %>% mutate(amnd = case_when(
  amnd == TRUE ~ "P",
  amnd == FALSE ~ "noP"
)) %>% group_by(Contig, time, amnd) %>% summarise(TPM = sum(TPM)) %>% pivot_wider(names_from = c(time, amnd), values_from = "TPM") %>% column_to_rownames("Contig")
cols <- colorRampPalette(brewer.pal(n = 9, name = "BrBG"))(100)
ann <- tmp2_mat %>% rownames_to_column("Contig") %>% left_join(tax) %>% left_join(hosts) %>% select(Contig, Class, host) %>% column_to_rownames("Contig") %>% mutate(Class = factor(Class), host = factor(host))
ann_colors = list(
    host = c(Bacteria = "#6fa8dc", Invertebrates = "#ffe599", Plants = "#b6d7a8", Fungi = "#ea9999", Eukaryotes = "#bcbcbc"),
    Class = c(Howeltoviricetes = "#FFFF99", Leviviricetes = "#6A3D9A", Lenarviricota = "#CAB2D6", Miaviricetes = "#FF7F00", Amabiliviricetes = "#FDBF6F", Tolucaviricetes = "#E31A1C", Duplopiviricetes = "#FB9A99", Alsuviricetes = "#33A02C", Kitrinoviricota = "#B2DF8A", Ellioviricetes = "#1F78B4", Pisoniviricetes = "#A6CEE3", Chunqiuviricetes = "#B15928", Chrymotiviricetes = "black", Unassigned = "#bcbcbc", Magsaviricetes = "purple", Stelpaviricetes = "yellow", Negarnaviricota = "navy", Pisuviricota = "cyan", Insthoviricetes = "lavender")
)

pheatmap(tmp2_mat, color = cols, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, legend = TRUE, show_rownames = FALSE, width = 8, na_col = "white", annotation_row = ann, annotation_colors = ann_colors, filename = "Pvotus_heatmap.pdf")
```

```{r}
# Rerun Permanova on Leviviricetes only
temp <- wetup_tax %>% filter(Class %like% "Levivi") %>%
  column_to_rownames("Contig") %>%
  select(starts_with("T", ignore.case = FALSE))
# toadd <- temp[, 1:3]
# colnames(toadd) <- gsub("A$", "P.A", colnames(toadd))
# colnames(toadd) <- gsub("B$", "P.B", colnames(toadd))
# colnames(toadd) <- gsub("C$", "P.C", colnames(toadd))
# temp <- data.frame(cbind(toadd, temp))
tempt <- data.frame(t(temp))
adonis2(tempt~time*Phos, data=design)

# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = tempt ~ time * Phos, data = design)
#           Df SumOfSqs      R2       F Pr(>F)    
# time       1   0.4931 0.05864  3.0237  0.001 ***
# Phos       1   1.9369 0.23036 11.8778  0.001 ***
# time:Phos  1   0.2709 0.03221  1.6610  0.058 .  
# Residual  35   5.7074 0.67879                   
# Total     38   8.4082 1.00000                    
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


```

```{r}
# Histogram of number of samples per vOTU
temp <- wetup_tax %>% select(-Order) %>%
  pivot_longer(cols = !c(Contig, taxonomy, Class), names_to = "sample", values_to = "count") %>%
  mutate(count = case_when(
    count > 0 ~ 1,
    TRUE ~ 0
  )) %>%
  group_by(Contig, Class) %>% summarise_at("count", sum) %>%
  mutate(Class = case_when(
    Class %in% top11 ~ Class,
    TRUE ~ "Other"
   )) #%>%
  # mutate(Class = factor(Class, levels=c(top11, "Other")))

ggplot(data=temp, aes(fill=Class, x = count)) + geom_histogram(stat="bin", binwidth = 1) + 
  scale_fill_brewer(palette="Paired", direction=-1) + theme_classic() + xlab("Number of samples") + ylab("Number of vOTUs")
ggsave("vOTUs_shared_hist.pdf", width=8.5, height = 11)

# Taxonomic profiles of vOTUs shared over X samples
temp_dis <- temp %>%
  group_by(Class, count) %>%
  tally %>%
  arrange(count, desc(n))  %>%
  mutate(rank = 1)
for (c in 1:39) {
  temp_dis$rank[temp_dis$count == c] <- 1:nrow(temp_dis[temp_dis$count == c,])
}
temp_dis_for_cols <- temp_dis %>% group_by(Class) %>% summarise(mean_rank = mean(rank)) %>% arrange(by = mean_rank)
temp_dis$Class <- factor(temp_dis$Class, levels = temp_dis_for_cols$Class)

cols <- c(Howeltoviricetes = "#FFFF99", Leviviricetes = "#6A3D9A", Lenarviricota = "#CAB2D6", Miaviricetes = "#FF7F00", Amabiliviricetes = "#FDBF6F", Tolucaviricetes = "#E31A1C", Duplopiviricetes = "#FB9A99", Alsuviricetes = "#33A02C", Kitrinoviricota = "#B2DF8A", Ellioviricetes = "#1F78B4", Pisoniviricetes = "#A6CEE3", Chunqiuviricetes = "#B15928", Chrymotiviricetes = "black", Unassigned = "#bcbcbc", Magsaviricetes = "purple", Stelpaviricetes = "yellow", Negarnaviricota = "navy", Pisuviricota = "cyan", Insthoviricetes = "lavender")

ggplot(data=temp_dis, aes(x = Class, y = rank, fill = Class)) + 
  theme_classic() + geom_boxplot() + xlab("Phylum") +  ylab("Rank") + 
  scale_fill_manual(values = cols) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ggsave("sup_fig_class_mean_rankabun.pdf", width=8.5, height = 11)  
```

```{r}
# Compare phylum level distributions
ictv <- read.delim("ICTV_RNA_viruses.txt", sep="\t", header=T, stringsAsFactors = F)

getPhylum <- function(taxname) {
  if (taxname %like% "ota$") return(taxname) # if it's already a phylum or class name keep it as is
  if (taxname == "Levivirales") return("Lenarviricota") # Levivirales don't exist anymore so I had to define this manually
  if (taxname == "Ophioviridae") return("Negarnaviricota") # Ophioviridae is a new family so I had to define this manually
  if (taxname %in% c("Zhaovirus", "Yanvirus", "Permutotetraviridae")) return("Unassigned") # These exceptions don't have a class or phylum
  if (taxname %like% "tes$") return(unique(ictv$Phylum[ictv$Class == taxname]))
  if (taxname %like% "les$") return(unique(ictv$Phylum[ictv$Order == taxname]))
  if (taxname %like% "dae$") return(unique(ictv$Phylum[ictv$Family == taxname]))
  else return(taxname)
}

wetup_tax$Phylum <- unlist(sapply(wetup_tax$Class, getPhylum))

# Convert numbers to percentages
wetup_pct <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Phylum, Class, Order), names_to="variable", values_to="tpm") %>%
  group_by(Phylum, variable) %>%
  summarize_at("tpm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = tpm*100/sum(tpm)) %>%
  mutate(Phylum = factor(Phylum))

# Plot mean abundance by phylum
wetup_pct <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Phylum, Class, Order), names_to="variable", values_to="tpm") %>%
  group_by(Phylum, variable) %>%
  summarize_at("tpm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = tpm*100/sum(tpm))

P4 <- ggplot(data = wetup_pct, aes(x = Phylum, y = pct, fill=Phylum)) + 
  geom_boxplot() + 
  scale_fill_brewer(palette = "Pastel2", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("taxonomy") + ylab("Relative abundance per sample")
ggsave(P4, "wetup_relabun_tax_box_phylum.pdf")

# Figure 3
ggarrange(P4, P1, P3, P2, ncol = 2, nrow = 2)
ggsave("Fig_3_taxonomy.pdf", width = 8, height = 8)

# Plot temporal abundance by phylum
# Mean and SE by Phylum
mean_ses <- wetup_pct %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", variable))) %>%
  mutate(amend = case_when(
    variable %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  group_by(Phylum, time) %>%
  mutate(pct_mean = mean(pct), pct_sd = sd(pct)) %>%
  mutate(sd_div_mean = pct_sd/pct_mean) %>%
  arrange(desc(pct_mean))

ggplot(mean_ses, aes(x = pct_mean, y = sd_div_mean)) + geom_point() + scale_x_continuous(trans='log2') +
  geom_smooth(method = lm) + stat_regline_equation(formula = y ~ x) + stat_cor(label.y=-0.1) + theme_classic()
# sd becomes a smaller % of the mean as the mean grows

# Temporal dynamics by phylum for viruses
df <- wetup_tax %>% column_to_rownames("Contig") %>% pivot_longer(cols=starts_with("T", ignore.case = FALSE), names_to = "sample", values_to = "tpm") %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", sample))) %>%
  mutate(amend = case_when(
    sample %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  mutate(replicate = gsub("T.*\\.", "", sample))
wP <- df %>% filter(amend == "P" | time == 0) %>% group_by(Phylum, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(tpm_mean = mean(tpm_sum), sd = sd(tpm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))
woP <- df %>% filter(amend == "noP") %>% group_by(Phylum, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(tpm_mean = mean(tpm_sum), sd = sd(tpm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))

pal <- rev(brewer.pal(6, "Blues"))
PP <- ggplot(data=wP, aes(x=time, y=tpm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + 
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
PnoP <- ggplot(data=woP, aes(x=time, y=tpm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + 
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1) + ylim(0, 1000000)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("tpm_sum_byphylum_line.pdf", width = 8, height = 4)
tpm_wohost <- sum(df$tpm)

l_R <- list()
l_p <- list()
l_perm <- list()
phyla <- c("Duplornaviricota", "Pisuviricota", "Lenarviricota", "Kitrinoviricota", "Negarnaviricota")
for (i in 1:5) {
  p <-  phyla[i]
  temp <- wetup_tax %>% filter(Phylum ==  p) %>% select(!c("Phylum", "taxonomy")) %>% column_to_rownames("Contig") %>% t %>% as.data.frame()
  temp <- temp[rowSums(temp) > 0,]
  wetup_tax_dist <- vegdist(temp, method = "bray")
  temp_design <- design[design$Sample %in% rownames(temp),]
  templ <- adonis2(wetup_tax_dist~time*Phos, data=temp_design)
  l_perm[[i]] <- templ
  l_R[[i]] <- templ$R2
  l_p[[i]] <- templ$`Pr(>F)`
}
# correct p values for multiple comparisons (3 per phylum * 5 phylum)
getsig <- function (x) {p.adjust(x, method="fdr", n=15)}
l_p <- lapply(l_p, getsig)
getsig <- function (x) {which(x <= 0.05)}
sigp <- lapply(l_p, getsig) # get indices of significant effects per phylum
sigR <- map2(l_R, sigp, `[`) # get R^2 values of these effects
# dsDNA: Duplorna unaffected, Pisu affected by P and time (6%, 5%). ssRNA+: Lenar by P and time (6%, 17%), Kitrino unaffected. ssRNA-: Negarna by P (10%).

# Is the TPM increase in Lenar significant?
temp <- wetup_tax %>% filter(Phylum %like% "Lenar") %>% select(!c("Phylum", "taxonomy")) %>% column_to_rownames("Contig") %>% pivot_longer(cols = everything(), names_to = "sample") %>% mutate(amnd = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
)) %>% mutate(time_point = gsub("T", "", gsub("\\..*", "", sample))) %>% mutate(repl = gsub(".*\\.", "", sample)) %>% group_by(time_point, amnd, repl) %>% summarise(tpm = sum(value))
temp_P <- temp %>% filter(time_point == "0" | amnd == "P")
temp_noP <- temp %>% filter(amnd == "noP")
dunn.test(temp_P$tpm, temp_P$time_point, alpha = 0.1, method = "bh") # T0 to T2
dunn.test(temp_noP$tpm, temp_noP$time_point, alpha = 0.1, method = "bh") # T0 to T2
```

```{r}
# Host prediction
hosts <- read.delim(file = "Trees/votus_w_host_assignment_uniq.txt", sep=" ", header=F)
colnames(hosts) <- c("Contig", "host")
for_pie <- hosts %>% group_by(host) %>% tally() %>% arrange(desc(n)) %>% mutate(host = factor(host, levels = c("Bacteria", "Fungi", "Invertebrates", "Plants", "Eukaryotes", "Mammals", "Sauria")))
# for_pie_n <- for_pie$n
# names(for_pie_n) <- rownames(for_pie)
# pie(for_pie_n, col = brewer.pal(length(for_pie_n), "Pastel2"))
ggplot(data=for_pie, aes(x = host, y=n, fill=host)) + geom_col(position = "stack") + theme_classic() + 
  scale_fill_brewer(palette = "Pastel1") + ylab("Number of vOTUs with assigned host group") + xlab("Host group")
ggsave("RNA_virs_host_dist.pdf")

# Temporal dynamics by phylum for viruses with an assigned host
wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]

df <- wetup_tax %>% left_join(hosts, by="Contig") %>% filter(!is.na(host))
df <- df %>% column_to_rownames("Contig") %>% pivot_longer(cols=starts_with("T", ignore.case = FALSE), names_to = "sample", values_to = "tpm") %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", sample))) %>%
  mutate(amend = case_when(
    sample %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  mutate(replicate = gsub("T.*\\.", "", sample))
tpm_hosts <- sum(df$tpm)
wP <- df %>% filter(amend == "P" | time == 0) %>% group_by(Phylum, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(tpm_mean = mean(tpm_sum), sd = sd(tpm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))
woP <- df %>% filter(amend == "noP") %>% group_by(Phylum, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(tpm_mean = mean(tpm_sum), sd = sd(tpm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))

pal <- rev(brewer.pal(6, "Blues"))
PP <- ggplot(data=wP, aes(x=time, y=tpm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative TPM") +  ylim(c(0, 1000000)) +
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
PnoP <- ggplot(data=woP, aes(x=time, y=tpm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") + ylim(c(0, 1000000)) +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + 
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("tpm_sum_byphylum_line.pdf", width = 8, height = 4)

# Is the increase in Lenar over time significant?
wP <- df %>% filter(amend == "P" | time == 0) %>% group_by(Phylum, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% filter(Phylum %like% "Lenar")
woP <- df %>% filter(amend == "noP") %>% group_by(Phylum, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% filter(Phylum %like% "Lenar")
dunn.test(wP$tpm_sum, wP$time, method = "bh", alpha = 0.1) # T0 lower than T2 and T3
dunn.test(woP$tpm_sum, woP$time, method = "bh", alpha = 0.1) # T0 lower than T1

# Without Lenarviricota
pal <- rev(brewer.pal(5, "Blues"))
temp_wP <- wP %>% filter(!(Phylum %like% "Lenar"))
temp_woP <- woP %>% filter(!(Phylum %like% "Lenar"))
PP <- ggplot(data=temp_wP, aes(x=time, y=tpm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + ylim(0, 100000) +
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
PnoP <- ggplot(data=temp_woP, aes(x=time, y=tpm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + ylim(0, 100000) +
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("tpm_sum_byhost_nolenar_line.pdf", width = 8, height = 4)

# Temporal dynamics by host
df <- wetup_tax %>% left_join(hosts, by="Contig") %>% filter(!is.na(host))
df <- df %>% column_to_rownames("Contig") %>% pivot_longer(cols=starts_with("T", ignore.case = FALSE), names_to = "sample", values_to = "tpm") %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", sample))) %>%
  mutate(amend = case_when(
    sample %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  mutate(replicate = gsub("T.*\\.", "", sample))
wP <- df %>% filter(amend == "P" | time == 0) %>% group_by(host, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% 
  group_by(host, time, amend) %>% summarise(tpm_mean = mean(tpm_sum), sd = sd(tpm_sum))
woP <- df %>% filter(amend == "noP") %>% group_by(host, time, amend, replicate) %>% summarise(tpm_sum = sum(tpm)) %>% 
  group_by(host, time, amend) %>% summarise(tpm_mean = mean(tpm_sum), sd = sd(tpm_sum))

pal <- c(Bacteria = "#6fa8dc", Fungi = "#ea9999", Invertebrates = "#ffe599", Sauria = "#8e7cc3", Mammals = "#783f04", Plants = "#b6d7a8", Other = "#bcbcbc")
HP <- ggplot(data=wP, aes(x=time, y=tpm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + ylim(c(0, 1000000)) +
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
HnoP <- ggplot(data=woP, aes(x=time, y=tpm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") + ylim(c(0, 1000000)) +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + 
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
ggarrange(PP, PnoP, HP, HnoP, nrow=2, ncol=2)
ggsave("tpm_sum_byhost_line.pdf", width = 8, height = 8)

pal <- c(Bacteria = "#6fa8dc", Fungi = "#ea9999", Invertebrates = "#ffe599", Sauria = "#8e7cc3", Mammals = "#783f04", Plants = "#b6d7a8", Other = "#bcbcbc")
temp_wP <- wP %>% filter(!(host %like% "Bacteria"))
temp_woP <- woP %>% filter(!(host %like% "Bacteria"))
PP <- ggplot(data=temp_wP, aes(x=time, y=tpm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") + ylim(0, 160000) +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + 
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
PnoP <- ggplot(data=temp_woP, aes(x=time, y=tpm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") + ylim(0, 160000) +
  xlab("Time (weeks)") + ylab("Cummulative TPM") + 
  geom_errorbar(aes(ymin=tpm_mean-sd, ymax=tpm_mean+sd), width=0.1)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("tpm_sum_byhost_nophages_line.pdf", width = 8, height = 4)

# Looks like there's high similarity between phylum level dynamics and dynamics of viruses with known hosts:
# Lenarviricota and phages
# Kitrinoviricota and fungal viruses
sum(df$tpm[(df$host=="Bacteria")]) / sum(df$tpm) # 85% of tpm of vOTUs with assigned host maps to phages
sum(df$tpm[(df$host=="Bacteria") & (df$Phylum=="Lenarviricota")]) / sum(df$tpm) # 83% of tpm of vOTUs with assigned host maps to Lenar phages
sum(df$tpm[(df$host=="Fungi")]) / sum(df$tpm) # 12.4% of tpm of vOTUs with assigned host maps to fungal viruses
sum(df$tpm[(df$host=="Fungi") & (df$Phylum=="Kitrinoviricota")]) / sum(df$tpm) # 10% of tpm of vOTUs with assigned host maps to Kitrino fungal viruses

# The tpm of viruses with hosts is 46% of tpm of total viruses with assigned phylum
tpm_hosts / tpm_wohost # 0.462

df2 <- df %>% select(host, Phylum, tpm) %>% 
  mutate(tpm = round(tpm, digits = 0)) %>%
  pivot_wider(names_from = host, values_from = tpm, values_fn = sum)
df2[is.na(df2)] <- 0
write.csv(df2, "hosts_by_phylum.csv")
```

```{r}
# Diversity and evenness indices
tax <- read.csv("Phylum_level_tax.tsv")

tax <- tax %>%
  select(Contig, Phylum) %>%
  unique()

wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]

# Diversity of viral communities over time after wetup

# Set palette colors
pal <- rev(brewer.pal(5, "Paired"))
# Shannon-Weaver diversity
H <- data.frame(diversity(t(wetup), "shannon")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(H)[2] <- "Shannon_div"
p_H <- ggplot(H, aes(x=treats, y=Shannon_div, fill=treats)) + 
  geom_boxplot() + 
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) +
  theme_bw() + ylab("Shannon diversity") + theme(axis.title.x=element_blank())
# Simpson index
simp <- data.frame(diversity(t(wetup), "simpson")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(simp)[2] <- "Simpson_div"
p_simp <- ggplot(simp, aes(x=treats, y=Simpson_div, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) + theme(legend.position="none") +
  theme_bw() + ylab("Simpson diversity") + theme(axis.title.x=element_blank())
# Inverse Simpson index
invsimp <- data.frame(diversity(t(wetup), "inv")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(invsimp)[2] <- "Inv_Simpson_div"
p_invsimp <- ggplot(invsimp, aes(x=treats, y=Inv_Simpson_div, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) + theme(legend.position="none") +
  theme_bw() + ylab("Inverse Simpson diversity") + theme(axis.title.x=element_blank())
# Species number
S <- data.frame(specnumber(t(wetup))) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(S)[2] <- "Spec_num"
p_S <- ggplot(S, aes(x=treats, y=Spec_num, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Species number") + theme(axis.title.x=element_blank())

# Evenness of viral communities over time after wetup
evenness <- data.frame(cbind(calcDiv(sampleData=t(wetup), type="PielouEven")), S$Sample)
evenness <- data.frame(evenness, S$treats)
colnames(evenness)[2:3] <- c("Sample", "treats")
evenness$Time <- gsub("T", "", gsub("\\..*", "", evenness$Sample))
p_E <- ggplot(evenness, aes(x=treats, y=PielouEven, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Pielou's evenness") + xlab("Treatment groups")

ggarrange(p_H, p_simp, p_invsimp, p_S, p_E, nrow=5, ncol=1, common.legend = TRUE)
ggsave("diversity_indices.pdf", width=8.5, height=10)

# Statistical analysis and p value adjustment for multiple comparisons
kruskal.test(Shannon_div~treats, data = H) # p=0.0002, padj=0.001
kruskal.test(Simpson_div~treats, data = simp) # p=0.001, padj=0.004
kruskal.test(Inv_Simpson_div~treats, data = invsimp) # p=0.001, padj=0.004
kruskal.test(Spec_num~treats, data = S) # p=0.2
kruskal.test(PielouEven~treats, data = evenness) # p=0.0002, padj=0.001
p.adjust(c(0.0001918, 0.001335, 0.001335, 0.2127, 0.0002018))

# Plot without isotope
H <- H %>% mutate(treats = gsub("18", "", treats))
S <- S %>% mutate(treats = gsub("18", "", treats))
evenness <- evenness %>% mutate(treats = gsub("18", "", treats))
p_H <- ggplot(H, aes(x=treats, y=Shannon_div, fill=treats)) + 
  geom_boxplot() + 
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) +
  theme_bw() + ylab("Shannon diversity") + theme(axis.title.x=element_blank())
p_S <- ggplot(S, aes(x=treats, y=Spec_num, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Species number") + theme(axis.title.x=element_blank())
p_E <- ggplot(evenness, aes(x=treats, y=PielouEven, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Pielou's evenness") + xlab("Treatment groups")
ggarrange(ggarrange(p_H, p_S, p_E, common.legend = T, ncol = 1, nrow = 3), P, ncol = 2, nrow = 1, widths = c(0.3, 0.7))
ggsave("fig_3_indices_PCoA.pdf", width = 8, height = 6)
```

```{r}
# Plot respiration
resp <- read.delim("respiration.tsv", sep="\t", header=T)
#colnames(resp)
resp <- resp %>% 
  mutate(CO2_ppm = as.numeric(gsub(",", "\\.", CO2_ppm))) %>%
  mutate(time = case_when(
  sample_ID %like% "std" ~ NA,
  sample_ID %like% "blank" ~ 0,
  .default = as.numeric(gsub("W.*", "", gsub("^.*T", "", sample_ID)))
)) %>%
  mutate(amend = case_when(
    sample_ID %like% "P" ~ "P",
    .default = "No P"
  )) %>% 
  select(time, CO2_ppm, amend) %>% pivot_longer(cols = CO2_ppm, values_to = "CO2") %>%
  filter(!is.na(time)) %>%
  group_by(time, amend) %>% summarise(CO2_mean = mean(CO2), CO2_sd = sd(CO2))
ggplot(data=resp, aes(x = time, y = CO2_mean, group = amend, color = amend)) + geom_point() + geom_line() + theme_classic() + 
  xlab("Time (weeks)") + ylab("CO2 (ppm)") + geom_errorbar(aes(ymin=CO2_mean-CO2_sd, ymax=CO2_mean+CO2_sd), width=.2, position=position_dodge(0.1))
ggsave("RNA_virus_respiration.pdf", width=8)

# Statistical testing
resp <- read.delim("respiration.tsv", sep="\t", header=T) %>% 
  mutate(CO2_ppm = as.numeric(gsub(",", "\\.", CO2_ppm))) %>%
  mutate(time = case_when(
  sample_ID %like% "std" ~ NA,
  sample_ID %like% "blank" ~ 0,
  .default = as.numeric(gsub("W.*", "", gsub("^.*T", "", sample_ID)))
)) %>%
  mutate(amend = case_when(
    sample_ID %like% "P" ~ "P",
    .default = "No P"
  )) %>% 
  select(time, CO2_ppm, amend) %>% pivot_longer(cols = CO2_ppm, values_to = "CO2") %>%
  filter(!is.na(time))
resp_P <- resp %>% filter(time==0 | amend=="P")
resp_noP <- resp %>% filter(amend=="No P")
pairwise.wilcox.test(resp_P$CO2, resp_P$time) # 0 is significanly different from all time points p<0.05
pairwise.wilcox.test(resp_noP$CO2, resp_noP$time) # 0 is significanly different from all time points p<0.05
```

```{r}
# Simulate infection by RNA phages during week 1
mapping_stats <- read.delim("SBB/table_S1_viralRNA.csv", header=T, sep=";")
mapping_stats <- mapping_stats %>%
  mutate(variable = paste(gsub("-", "", Time_Point), ".R.", gsub("_", ".", Sample_Name), sep=""))
# select phages and phage clades
wetup_tax <- read.delim("table_S4_vOTUs.txt", sep = "\t", header = T)
phg <- wetup_tax %>% filter(Class %like% "Levivi" | Order %like% "Durna" | host %like% "Bacteria") %>%
  select(starts_with("T", ignore.case = F)) %>% pivot_longer(cols = everything(), names_to = "variable") %>% 
  group_by(variable) %>% summarise(tpm = sum(value)) %>% left_join(mapping_stats, by = "variable")
# RNA was extracted from 0.5 g soil, so I'm multiplying the cell number by 2 to normalize per g
phg <- phg %>% 
  mutate(phg_pct = tpm/num_reads) %>%
  mutate(phg_RNA_ng = phg_pct * 2 * Total.metaT.RNA..ng.) %>%
  mutate(Time_Point = gsub("T-", "", Time_Point))

phg_P <- phg %>% filter(Time_Point=="0" | str_detect(variable, "P"))
phg_noP <- phg %>% filter(str_detect(variable, "P", negate = TRUE))
durna_P <- durna %>% filter(Time_Point=="0" | str_detect(variable, "P"))
durna_noP <- durna %>% filter(str_detect(variable, "P", negate = TRUE))
# Panels A and B
phg_P_fig <- ggplot(phg_P, aes(x = Time_Point, y = phg_RNA_ng)) + geom_boxplot(fill = "purple") + theme_classic() + 
  xlab("Time (weeks)") + ylab("RNA from Leviviricetes (ng / g soil)") + ggtitle("With P") + ylim(c(0, 40))
phg_noP_fig <- ggplot(phg_noP, aes(x = Time_Point, y = phg_RNA_ng)) + geom_boxplot(fill = "salmon") + theme_classic() +
  xlab("Time (weeks)") + ylab("RNA from Leviviricetes (ng / g soil)") + ggtitle("Without P") + ylim(c(0, 40))

# Statistical testing of RNA amount over time
dunn.test(x = phg_P$phg_RNA_ng, g = phg_P$Time_Point, method="bh") # T0 to T2: 0.005, T0 to T3: 0.02
dunn.test(x = phg_noP$phg_RNA_ng, g = phg_noP$Time_Point, method="bh") # T0 to T1: 0.03, T0 to T2: 0.02

phg_P_mean <- phg_P %>% select(Time_Point, phg_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(phg_RNA_ng_mean = round(mean(phg_RNA_ng), 2), phg_RNA_ng_sd  = round(sd(phg_RNA_ng), 2)) %>%
  mutate(phg_RNA_ng_diff = phg_RNA_ng_mean - phg_RNA_ng_mean[1])
phg_noP_mean <- phg_noP %>% select(Time_Point, phg_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(phg_RNA_ng_mean = round(mean(phg_RNA_ng), 2), phg_RNA_ng_sd  = round(sd(phg_RNA_ng), 2)) %>%
  mutate(phg_RNA_ng_diff = phg_RNA_ng_mean - phg_RNA_ng_mean[1])
durna_P_mean <- durna_P %>% select(Time_Point, durna_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(durna_RNA_ng_mean = round(mean(durna_RNA_ng), 5), durna_RNA_ng_sd  = round(sd(durna_RNA_ng), 2)) 
durna_noP_mean <- durna_noP %>% select(Time_Point, durna_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(durna_RNA_ng_mean = round(mean(durna_RNA_ng), 5), durna_RNA_ng_sd  = round(sd(durna_RNA_ng), 2)) 
# The average Leviviricetes genome size is 4030 bp, weighing on average 1.31*10^-12 ng
# We can back-calculate how many phages the RNA from Leviviricetes represents 
# By picking a burst size we can calculate how many cells lysed on average
# RNA was extracted from 0.5 g soil, so I'm multiplying the cell number by 2 to normalize per g
# The amount of RNA per genome of viruses varies by phylum, but since most are Lenarviricota (ssRNA), 
# this approximation will only consider a single strand
phg_P_mean <- phg_P_mean %>% 
  mutate(num_virions = phg_RNA_ng_diff*2/(1.31e-12)) %>%
  mutate(num_cells_BS_200 = num_virions/200) %>%
  mutate(num_cells_BS_2000 = num_virions/2000) %>%
  mutate(num_cells_BS_20000 = num_virions/20000) %>%
  mutate(num_cells_BS_200000 = num_virions/200000) %>%
  pivot_longer(cols = starts_with("num_cells"), names_to = "burst_size", values_to = "num_cells") %>%
  mutate(burst_size =  gsub(".*_", "", burst_size))
phg_noP_mean <- phg_noP_mean %>% 
  mutate(num_virions = phg_RNA_ng_diff*2/(1.31e-12)) %>%
  mutate(num_cells_BS_200 = num_virions/200) %>%
  mutate(num_cells_BS_2000 = num_virions/2000) %>%
  mutate(num_cells_BS_20000 = num_virions/20000) %>%
  mutate(num_cells_BS_200000 = num_virions/200000) %>%
  pivot_longer(cols = starts_with("num_cells"), names_to = "burst_size", values_to = "num_cells") %>%
  mutate(burst_size =  gsub(".*_", "", burst_size))

# Panels C and D
phg_P_mean_fig <- ggplot(data = phg_P_mean, aes(x = Time_Point, y = num_cells+1, colour = burst_size, group = burst_size)) +
  theme_classic() + xlab("Time (weeks)") + ylab("Mean total virocells / g soil") +
  scale_color_brewer(palette = "Purples", direction = -1) + geom_point() + geom_line() + 
  ggtitle("With P") + scale_y_continuous(trans='log10')
phg_noP_mean_fig <- ggplot(data = phg_noP_mean, aes(x = Time_Point, y = num_cells+1, colour = burst_size, group = burst_size)) +
  theme_classic() + xlab("Time (weeks)") + ylab("Mean total virocells / g soil") +
  scale_color_brewer(palette = "Reds", direction = -1) + geom_point() + geom_line() + 
  ggtitle("Without P") + scale_y_continuous(trans='log10')

ggarrange(phg_P_fig, phg_noP_fig, phg_P_mean_fig, phg_noP_mean_fig, nrow = 2, ncol = 2)
ggsave("fig5_lysis_simulation.pdf", height=6, width=8)
```

```{r}
#Separate PCoA of phages (Leviviricetes and Durnavirales) and other viruses
tax <- read.csv("Order_level_tax.tsv", header = TRUE)
wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]

wetup_tax <- wetup %>% rownames_to_column("Contig") %>% left_join(tax, by="Contig")

phage_comm <- wetup_tax %>% filter(Class %like% "Levivi" | Order %like% "Durnavi")
other_comm <- wetup_tax %>% filter(!(Class %like% "Levivi" | Order %like% "Durnavi"))

temp <- phage_comm %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = FALSE))
wetup_sqrt <- sqrt(temp[rowSums(temp) > 0,])
wetup_dist <- vegdist(t(wetup_sqrt), method="bray", na.rm = T)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_")) %>%
  mutate(ellps = as.factor(gsub("T[1-3]_", "", cond)))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample, group=ellps)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  theme_bw()
ggsave("RNA_phages_PCoA_wetup_sqrt.pdf", width=10)

temp <- other_comm %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = FALSE))
wetup_sqrt <- sqrt(temp[rowSums(temp) > 0,])
wetup_dist <- vegdist(t(wetup_sqrt), method="bray", na.rm = T)

pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df$grp[pcoa_df$grp %like% "T0"] <- "T0_16O"
pcoa_df$grp[pcoa_df$grp %like% "H2O"] <- str_c(substr(pcoa_df$grp[pcoa_df$grp %like% "H2O"], 1, 2), "16O", sep = "_")
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_")) %>%
  mutate(ellps = as.factor(gsub("T[1-3]_", "", cond)))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample, group=ellps)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  theme_bw()
ggsave("RNA_nonphages_PCoA_wetup_sqrt.pdf", width=10, height = 7)
```

```{r}
# ITS taxonomy
its_otus <- read.delim("qiime2/rep-seqs_ITS2_SE_vs_NCBI_ITS_besthit.tsv", sep="\t", header=F)
colnames(its_otus) <- c("asv", "acc", "pid", "qcovs", "qstart", "qend", "sstart", "send", "length", "evalue", "bitscore") 
its_otus$acc <- gsub("\\|", "", gsub("[a-z]*\\|", "", its_otus$acc))
its_otus$acc <- gsub("\\..*", "", its_otus$acc)
its_otus$ncbi_taxa_id <- accessionToTaxa(its_otus$acc, 'accessionTaxa.sql', version = 'base')
its_otus$ncbi_tax <- getTaxonomy(its_otus$ncbi_taxa_id, 'accessionTaxa.sql')
its_otus <- its_otus %>% select("asv", "ncbi_taxa_id", starts_with("ncbi_tax"))
its_otus_for_table <- data.frame(its_otus$ncbi_tax)
its_otus_for_table <- data.frame(its_otus$asv, its_otus_for_table)
colnames(its_otus_for_table)[1] <- "asv"
write_delim(its_otus_for_table, "ITS2_NCBI_taxonomy.tsv", delim = "\t")

# You can just read the table since I made it
its_otus_for_table <- read.delim("qiime2/ITS2_NCBI_taxonomy.tsv", sep = "\t")

# Add ITS ASV table
euks <- read.delim("qiime2/ITS2_SE_ASV_table.tsv", header = T, sep = "\t")
colnames(euks)[1:3] <- c("asv", "T.0.Dry.Soil.A", "T.0.Dry.Soil.B")
euks <- euks %>% left_join(its_otus_for_table, by = "asv") %>% filter(!(is.na(phylum))) %>% filter(phylum %like% "mycota")
euk_melt <- euks %>% column_to_rownames("asv") %>% pivot_longer(cols = starts_with("T."), names_to = "sample") %>% mutate(time = substr(sample, 3, 3)) %>% mutate(treat = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
))
euks_for_dist <- euks %>% column_to_rownames("asv") %>% select(starts_with("T."))
colnames(euks_for_dist) <- gsub("T\\.", "T", colnames(euks_for_dist))

# Plot
ggplot(euk_melt, aes(x = sample, y = value, fill = phylum)) + geom_bar(position = "fill", stat = "identity", na.rm = T) + theme_bw() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_brewer(palette = "Paired") + xlab("Sample") + ylab("Proportion")
ggsave("Fungi.pdf")
temp <- euk_melt %>% filter(phylum %like% "Ascomycota") %>% filter(!(is.na(class)))
ggplot(temp, aes(x = sample, y = value, fill = class)) + geom_bar(position = "fill", stat = "identity", na.rm = T) + theme_bw() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_brewer(palette = "Paired") + xlab("Sample") + ylab("Proportion")
ggsave("Ascomycota.pdf")

# Statistics
euk_dist <- vegdist(t(euks_for_dist), method = "bray")
design_comm <- read.delim("design_comm.txt", sep = "\t", header = T)
adonis2(euk_dist~time*Phos, data=design_comm)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = euk_dist ~ time * Phos, data = design_comm)
#           Df SumOfSqs      R2      F Pr(>F)  
# time       1   0.2687 0.06689 2.0253  0.014 *
# Phos       1   0.1391 0.03463 1.0486  0.351  
# time:Phos  1   0.1597 0.03974 1.2033  0.236  
# Residual  26   3.4500 0.85873                
# Total     29   4.0176 1.00000                
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

euks_sqrt <- sqrt(euks_for_dist)
euks_dist <- vegdist(t(euks_sqrt), method="bray", na.rm = T)
euks_cols <- factor(design_comm$cosm)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(euks_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = substr(sample, 1, 2)) %>% 
  mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
p1 <- ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  theme_bw() + ggtitle("Fungi")
ggsave("RNA_viruses_PCoA_ITS2_sqrt.pdf", width=10, height = 7)

# Statistics of fungal viruses community dynamics
wetup_tax <- read.delim("table_S4_vOTUs.txt", sep = "\t", header = T)
myc <- wetup_tax %>% filter(host %like% "Fungi") %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = F))
myc_sqrt <- sqrt(myc)
myc_dist <- vegdist(t(myc), method = "bray")
design <- read.delim("mtx_design_Gary.txt")
adonis2(myc_dist~time*Phos, data = design)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = myc_dist ~ time * Phos, data = design)
#           Df SumOfSqs      R2      F Pr(>F)    
# time       1   0.5205 0.05690 2.3799  0.001 ***
# Phos       1   0.4575 0.05001 2.0918  0.002 ** 
# time:Phos  1   0.5146 0.05626 2.3528  0.001 ***
# Residual  35   7.6551 0.83683                  
# Total     38   9.1478 1.00000                  
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

myc_cols <- factor(design_comm$cosm)
myc_dist <- vegdist(t(myc_sqrt), method = "bray")

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(myc_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = substr(sample, 1, 3)) %>% 
  mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
p2 <- ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  theme_bw() + ggtitle("Mycoviruses")
ggsave("RNA_viruses_PCoA_mycovirs_sqrt.pdf", width=10, height = 7)

# Shannon-Weaver diversity
pal <- brewer.pal(name = "Paired", n=7)
H <- data.frame(diversity(t(euks_for_dist), "shannon")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("18", "", gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample))))
colnames(H)[2] <- "Shannon_div"
p_H_euk <- ggplot(H, aes(x=treats, y=Shannon_div, fill=treats)) + 
  geom_boxplot() + 
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) +
  theme_bw() + ylab("Shannon diversity") + theme(axis.title.x=element_blank())
# Species number
S <- data.frame(specnumber(t(euks_for_dist))) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("18", "", gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample))))
colnames(S)[2] <- "Spec_num"
p_S_euk <- ggplot(S, aes(x=treats, y=Spec_num, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Species number") + theme(axis.title.x=element_blank())

# Evenness of eukaryotic communities over time after wetup
evenness <- data.frame(cbind(calcDiv(sampleData=t(euks_for_dist), type="PielouEven")), S$Sample)
evenness <- data.frame(evenness, S$treats)
colnames(evenness)[2:3] <- c("Sample", "treats")
evenness$Time <- gsub("T", "", gsub("\\..*", "", evenness$Sample))
p_E_euk <- ggplot(evenness, aes(x=treats, y=PielouEven, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Pielou's evenness") + xlab("Treatment groups")

# without time as a parameter
# Shannon-Weaver diversity
pal <- rev(brewer.pal(name = "Paired", n=5))[1:3]
H <- data.frame(diversity(t(euks_for_dist), "shannon")) %>% rownames_to_column("Sample") %>%
  mutate(treats = case_when(
    str_detect(Sample, "Dry") ~ "Dry",
    str_detect(Sample, "P") ~ "P",
    .default = "noP"
  ))
colnames(H)[2] <- "Shannon_div"
p_H_euk <- ggplot(H, aes(x=treats, y=Shannon_div, fill=treats)) + 
  geom_boxplot() + 
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) +
  theme_bw() + ylab("Shannon diversity") + theme(axis.title.x=element_blank())
# Species number
S <- data.frame(specnumber(t(euks_for_dist))) %>% rownames_to_column("Sample") %>%
  mutate(treats = case_when(
    str_detect(Sample, "Dry") ~ "Dry",
    str_detect(Sample, "P") ~ "P",
    .default = "noP"
  ))
colnames(S)[2] <- "Spec_num"
p_S_euk <- ggplot(S, aes(x=treats, y=Spec_num, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Species number") + theme(axis.title.x=element_blank())

# Evenness of eukaryotic communities over time after wetup
evenness <- data.frame(cbind(calcDiv(sampleData=t(euks_for_dist), type="PielouEven")), S$Sample)
evenness <- data.frame(evenness, S$treats)
colnames(evenness)[2:3] <- c("Sample", "treats")
evenness$Time <- gsub("T", "", gsub("\\..*", "", evenness$Sample))
p_E_euk <- ggplot(evenness, aes(x=treats, y=PielouEven, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Pielou's evenness") + xlab("Treatment groups")
```

```{r}
# Dynamics of phages with iPHoP hosts
phages <- read.csv("iphop_genome_predictions_new.csv", header=T) %>% select(!c(Host.genome, Main.method)) %>% unique %>% group_by(Contig) %>% top_n(1, Confidence.score) %>% mutate(Host.taxonomy = gsub(";s.*", "", Host.taxonomy)) %>% unique()
phages_dyn <- wetup_tax %>% filter(Contig %in% phages$Contig) %>% mutate(virname = str_c("vOTU", row_number(), sep = "_")) %>% left_join(phages, by = "Contig")
data_cols  <- which(colnames(phages) %like% "^T[0-3]")
phages_dyn_rel <- phages_dyn
temp2 <- phages_dyn_rel[, data_cols] * 100 / colSums(wetup_tax[, data_cols])
phages_dyn_rel[, data_cols] <- temp2
phages_dyn_rel_melt <- phages_dyn_rel %>% pivot_longer(cols = matches("T[0-3]"), names_to = "sample") %>% mutate(time = substr(sample, 2, 2)) %>% mutate(treat = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
)) %>% mutate(host_gn = gsub("^.*g__", "", Host.taxonomy))

temp <- phages_dyn_rel
temp %>% mutate(Host.taxonomy = gsub(";o.*", "", Host.taxonomy)) %>% group_by(Host.taxonomy) %>% tally()
host_gn <- gsub("^.*g__", "", phages_dyn_rel$Host.taxonomy)

# Putative host dynamics
bacs <- read.delim("qiime2/phyloseq/otu_table_16S.tsv") %>% select(!NEGATIVE)
colnames(bacs)[2:3] <- c("T.0.Dry.Soil.A", "T.0.Dry.Soil.B")
colnames(bacs)[1] <- "Feature.ID"
bacs_tax <- read.delim("qiime2/taxonomy_16S.tsv") %>% select(!Confidence)
bacs <- bacs %>% left_join(bacs_tax, by = "Feature.ID")
bacs_rlvt <- bacs %>% mutate(genus = gsub(";.*", "", gsub("^.*g__", "", Taxon))) %>% filter(genus %in% host_gn) # only 10  ASVs  have a matching host genus

host_gn_uniq <- unique(bacs_rlvt$genus)
data_cols  <- which(colnames(bacs_rlvt) %like% "^T\\.")
tot_count <- colSums(bacs[, data_cols])
bacs_rlvt_rel <- bacs_rlvt
bacs_rlvt_rel[ ,data_cols] <- bacs_rlvt[ ,data_cols]*100 / tot_count
bacs_rlvt_rel <- bacs_rlvt_rel %>% pivot_longer(cols = starts_with("T."), names_to = "sample") %>% mutate(time = substr(sample,3,3)) %>% mutate(treat = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
))

for (gn in host_gn_uniq) {
  temp <- bacs_rlvt_rel %>% filter(genus == gn) %>% mutate(grp = str_c(Feature.ID, treat, sep = "_")) %>% group_by(time, treat) %>% summarise(tot = sum(value))
  my_pal = c(brewer.pal(12, "Paired"), "grey", "black")
  ggplot(data = temp, aes(x = time, y = tot, color = treat, group = treat)) + geom_point() + theme_bw() + geom_line(stat = "identity") + 
    scale_color_manual(values = my_pal)
  ggsave(str_c(gn, ".pdf"))
}

# Statistics
colnames(bacs) <- gsub("X[1-2]", "T.0.", colnames(bacs))
bac_for_dist <- bacs %>% column_to_rownames("Feature.ID") %>% select(starts_with("T."))
colnames(bac_for_dist) <- gsub("T\\.", "T", colnames(bac_for_dist))
bac_dist <- vegdist(t(bac_for_dist), method = "bray")
design_comm <- read.delim("design_comm.txt", sep = "\t", header = T)
adonis2(bac_dist~time*Phos, data=design_comm)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = bac_dist ~ time * Phos, data = design_comm)
#           Df SumOfSqs      R2      F Pr(>F)
# time       1   0.3989 0.03291 0.9558  0.100
# Phos       1   0.4528 0.03735 1.0848  0.159
# time:Phos  1   0.4177 0.03446 1.0008  0.460
# Residual  26  10.8522 0.89528              
# Total     29  12.1216 1.00000

bacs_sqrt <- sqrt(bac_for_dist)
bacs_dist <- vegdist(t(bacs_sqrt), method="bray", na.rm = T)
bacs_cols <- factor(design_comm$cosm)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(bacs_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = gsub("\\.", "", substr(sample, 1, 3))) %>% 
  mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
p3 <- ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  theme_bw() + ggtitle("Prokaryotes")
ggsave("RNA_viruses_PCoA_16S_sqrt.pdf", width=10, height = 7)

# Statistics of phage community dynamics
wetup_tax <- read.delim("table_S4_vOTUs.txt", sep = "\t", header = T)
phg <- wetup_tax %>% filter(Class %like% "Levivi" | Order %like% "Durna" | host %like% "Bacteria") %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = F))
phg_sqrt <- sqrt(phg)
phg_dist <- vegdist(t(phg), method = "bray")
adonis2(phg_dist~time*Phos, data = design)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = phg_dist ~ time * Phos, data = design)
#           Df SumOfSqs      R2       F Pr(>F)    
# time       1   0.5658 0.09172  5.0118  0.001 ***
# Phos       1   1.3466 0.21828 11.9282  0.001 ***
# time:Phos  1   0.3054 0.04951  2.7053  0.009 ** 
# Residual  35   3.9512 0.64049                   
# Total     38   6.1690 1.00000                   
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

phg_cols <- factor(design_comm$cosm)
phg_dist <- vegdist(t(phg_sqrt), method = "bray")

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(phg_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = substr(sample, 1, 2)) %>% 
  mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
p4 <- ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  theme_bw() + ggtitle("Phages")
ggsave("RNA_viruses_PCoA_phages_sqrt.pdf", width=10, height = 7)

ggarrange(p3, p1, p4, p2, ncol = 2, nrow = 2, common.legend = T)
ggsave("fig6_hostdyn_pcoa.pdf", height = 6, width = 8)
```
```{r}
# Diversity indices for bacteria
# Shannon-Weaver diversity
pal <- brewer.pal(name = "Paired", n=7)
H <- data.frame(diversity(t(bac_for_dist), "shannon")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("18", "", gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample))))
colnames(H)[2] <- "Shannon_div"
p_H_bac <- ggplot(H, aes(x=treats, y=Shannon_div, fill=treats)) + 
  geom_boxplot() + 
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) +
  theme_bw() + ylab("Shannon diversity") + theme(axis.title.x=element_blank())
# Species number
S <- data.frame(specnumber(t(bac_for_dist))) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("18", "", gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample))))
colnames(S)[2] <- "Spec_num"
p_S_bac <- ggplot(S, aes(x=treats, y=Spec_num, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Species number") + theme(axis.title.x=element_blank())

# Evenness of eukaryotic communities over time after wetup
evenness <- data.frame(cbind(calcDiv(sampleData=t(bac_for_dist), type="PielouEven")), S$Sample)
evenness <- data.frame(evenness, S$treats)
colnames(evenness)[2:3] <- c("Sample", "treats")
evenness$Time <- gsub("T", "", gsub("\\..*", "", evenness$Sample))
p_E_bac <- ggplot(evenness, aes(x=treats, y=PielouEven, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Pielou's evenness") + xlab("Treatment groups")

ggarrange(p_H_euk, p_H_bac, p_S_euk, p_S_bac, p_E_euk, p_E_bac, nrow = 3, ncol = 2, common.legend = T)
ggsave("sup_fig_diversity_euk_bac_bytime.pdf", height = 10, width = 8)

# Without time as a parameter
# Shannon-Weaver diversity
pal <- rev(brewer.pal(name = "Paired", n=5))[1:3]
H <- data.frame(diversity(t(bac_for_dist), "shannon")) %>% rownames_to_column("Sample") %>%
  mutate(treats = case_when(
    str_detect(Sample, "Dry") ~ "DrySoil",
    str_detect(Sample, "P") ~ "P",
    .default = "noP"
  ))
colnames(H)[2] <- "Shannon_div"
p_H_bac <- ggplot(H, aes(x=treats, y=Shannon_div, fill=treats)) + 
  geom_boxplot() + 
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) +
  theme_bw() + ylab("Shannon diversity") + theme(axis.title.x=element_blank())
# Species number
S <- data.frame(specnumber(t(bac_for_dist))) %>% rownames_to_column("Sample") %>%
  mutate(treats = case_when(
    str_detect(Sample, "Dry") ~ "DrySoil",
    str_detect(Sample, "P") ~ "P",
    .default = "noP"
  ))
colnames(S)[2] <- "Spec_num"
p_S_bac <- ggplot(S, aes(x=treats, y=Spec_num, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Species number") + theme(axis.title.x=element_blank())

# Evenness of eukaryotic communities over time after wetup
evenness <- data.frame(cbind(calcDiv(sampleData=t(bac_for_dist), type="PielouEven")), S$Sample)
evenness <- data.frame(evenness, S$treats)
colnames(evenness)[2:3] <- c("Sample", "treats")
evenness$Time <- gsub("T", "", gsub("\\..*", "", evenness$Sample))
p_E_bac <- ggplot(evenness, aes(x=treats, y=PielouEven, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Pielou's evenness") + xlab("Treatment groups")

ggarrange(p_H_euk, p_H_bac, p_S_euk, p_S_bac, p_E_euk, p_E_bac, nrow = 3, ncol = 2, common.legend = T)
ggsave("sup_fig_diversity_euk_bac_bytime.pdf", height = 10, width = 8)
```

```{r}
#Viruses with an external phase
capsids <- scan("DRAM/contigs_w_capsid.txt", what="character()")
temp <- wetup_tax[wetup_tax$Contig %in% capsids,]
pcts <- colSums(temp[,2:40])*100/colSums(wetup_tax[,2:40]) # % of viruses with capsid gene out of total viruses
max(pcts) # 3.5%
min(pcts) # 0.8%
mean(pcts) # 1.3%

cols_P <- which(colnames(temp) %like% "P")
cols_noP <- which(!(colnames(temp) %like% "P") & !(colnames(temp) %like% "T0"))
cols_noP <- cols_noP[cols_noP < 40 & cols_noP > 1]
t.test(pcts[cols_P], pcts[cols_noP]) # no significant difference in relative abundance with P for all viruses together

temp_melt <- temp_num %>% rownames_to_column("Contig") %>% pivot_longer(cols = !Contig, names_to = "sample") %>% mutate(amnd = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
))
temp_comp <- compare_means(value ~ amnd, temp_melt, group.by = "Contig")

temp <- temp %>% left_join(hosts, by="Contig")
temp_num <- temp %>% remove_rownames() %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = F))
ann <- temp %>% remove_rownames() %>% column_to_rownames("Contig") %>% select(Order, host)
pheatmap(temp_num, color = colorRampPalette(brewer.pal(name="YlOrRd", n=9))(100), annotation_row = ann, cluster_rows = T, cluster_cols = F, show_rownames = F)
```

```{r}
# Differential abundance of 16S ASVs
colnames(bacs)[2:3] <- c("T.0.Dry.soil.A", "T.0.Dry.soil.B")
design_comm$time <- factor(design_comm$time)
design_comm$Sample <- gsub("R\\.", "", design_comm$Sample)
#Run DESeq2
bacs_cts <- bacs %>% column_to_rownames("Feature.ID") %>% select(starts_with("T."))
colnames(bacs_cts) <- gsub("T\\.", "T", colnames(bacs_cts))
bacs_cts <- bacs_cts[,colSums(bacs_cts) > 3000]
design_comm <- design_comm %>% filter(Sample %in% colnames(bacs_cts))
dds <- DESeqDataSetFromMatrix(countData = bacs_cts,
                              colData = design_comm,
                              design= ~ Phos + time)
dds <- DESeq(dds)
res <- results(dds)
resultsNames(dds)

dds_counts <- counts(dds, normalized=TRUE)
rownames(dds_counts) <- rownames(res)
dds_16S_tax <- merge(dds_counts, bacs_tax, by.x="row.names", by.y="row")
write.csv(dds_16S_tax, file = "dds_16S_tax.csv")
saveRDS(dds, file = "dds_16S_tax.rds")
dds <- readRDS("dds_16S_tax.rds")

colnames(bacs_tax)[1] <- "row"

# P vs. no P
res <- results(dds, tidy=TRUE, contrast=c("Phos", "P", "None")) %>%
  arrange(padj, pvalue) %>%
  tibble() %>% 
  left_join(bacs_tax, by = "row") %>%
  mutate(tax_short = gsub("^.*__", "", gsub("; [a-z]__$", "", gsub("; [a-z]__;.*", "", Taxon))))

EnhancedVolcano(res,
    lab = res$tax_short,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 10e-10,
    FCcutoff = 1,
    pointSize = 1.0,
    labSize = 1.5,
    shape = 1,
    title = "With vs. without P amendment")

sig_taxa <- res %>% filter(padj < 10e-10) %>% mutate(class = gsub(";.*", "", gsub("^.*c__", "", Taxon))) %>%
  mutate(phylum = gsub(";.*", "", gsub("^.*p__", "", Taxon)))
sig_taxa_down <- sig_taxa %>% filter(log2FoldChange < 0)
sig_taxa_up <- sig_taxa %>% filter(log2FoldChange > 0) # All Actinobacteria
ggplot(sig_taxa_down, aes(x = phylum, y = log2FoldChange)) + geom_boxplot(aes(fill = phylum)) + 
  theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_brewer(palette = "Set3")
ggsave("bacs_deseq_down_with_P.pdf", height = 9, width = 8.5)
n_per_phylum <- sig_taxa_down %>% group_by(phylum) %>% tally()
# Table S7
asvs <- bacs_cts %>% rownames_to_column("row") %>% filter(row %in% sig_taxa_down$row) %>% left_join(bacs_tax, by = "row")
write.table(asvs, "table_S7_16S_asvs.tsv", sep = ",", row.names = F)
# When did these asvs appear first?
asvs <- pivot_longer(asvs, cols = !c(row, Taxon), names_to = "sample", values_to = "tpm") %>% 
  mutate(grp = gsub("\\.[A-C]$", "", sample)) %>% mutate(presence = case_when(
    tpm > 0 ~ 1,
    .default = 0
  )) %>% mutate(grp = gsub("18", "", grp)) %>% 
  group_by(grp) %>% summarise(num_asvs = sum(presence))

# T1 vs. T0 - no significant results
res <- results(dds, tidy=TRUE, contrast=c("time", "1", "0")) %>%
  arrange(padj, pvalue) %>%
  tibble() %>% 
  left_join(bacs_tax, by = "row") %>%
  mutate(tax_short = gsub("^.*__", "", gsub("; [a-z]__$", "", gsub("; [a-z]__;.*", "", Taxon))))

EnhancedVolcano(res,
    lab = res$tax_short,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 10e-10,
    FCcutoff = 1,
    pointSize = 1.0,
    labSize = 1.5,
    shape = 1,
    title = "After one week compared to dry soil")
```

```{r}
# Differential abundance of ITS2 ASVs
design_comm$time <- factor(design_comm$time)
design_comm$Sample <- gsub("R\\.", "", design_comm$Sample)
#Run DESeq2
euks_cts <- euks %>% column_to_rownames("asv") %>% select(starts_with("T."))
colnames(euks_cts) <- gsub("T\\.", "T", colnames(euks_cts))
euks_cts <- euks_cts[,colSums(euks_cts) > 1800]
colnames(euks_cts) <- gsub("Soil", "soil", colnames(euks_cts))
design_comm <- design_comm %>% filter(Sample %in% colnames(euks_cts))
dds <- DESeqDataSetFromMatrix(countData = euks_cts,
                              colData = design_comm,
                              design= ~ Phos + time)
dds <- DESeq(dds)
res <- results(dds)
resultsNames(dds)

dds_counts <- counts(dds, normalized=TRUE)
rownames(dds_counts) <- rownames(res)
dds_ITS2_tax <- merge(dds_counts, its_otus_for_table, by.x="row.names", by.y="asv")
write.csv(dds_ITS2_tax, file = "dds_ITS2_tax.csv")
saveRDS(dds, file = "dds_ITS2_tax.rds")

colnames(its_otus_for_table)[1] <- "row"

# P vs. no P: only one ASV affected, Barriopsis
res <- results(dds, tidy=TRUE, contrast=c("Phos", "P", "None")) %>%
  arrange(padj, pvalue) %>%
  tibble() %>% 
  left_join(its_otus_for_table, by = "row")

EnhancedVolcano(res,
    lab = res$genus,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 10e-10,
    FCcutoff = 1,
    pointSize = 2.0,
    labSize = 3,
    shape = 1,
    title = "With vs. without P amendment")

# T1 vs. T0 - no significant results
res <- results(dds, tidy=TRUE, contrast=c("time", "1", "0")) %>%
  arrange(padj, pvalue) %>%
  tibble() %>% 
  left_join(its_otus_for_table, by = "row") 

EnhancedVolcano(res,
    lab = res$genus,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 10e-10,
    FCcutoff = 1,
    pointSize = 1.0,
    labSize = 1.5,
    shape = 1,
    title = "After one week compared to dry soil")
```

```{r}
# Use DESeq2 to identify vOTUs that bloomed over the first week or with P amendment
design_phage <- design %>% filter(time == 0 | time == 1) %>% mutate(time = as_factor(time))
design_phage$grp <- paste(design_phage$Phos, design_phage$time, sep="_")
#Run DESeq2
wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]
votus <- rownames(wetup)
samps <- design_phage$Sample

reads <- read.delim("coverm_count_breadth50_Gary.tsv", sep="\t", header=T) %>% filter(Contig %in% votus) %>% column_to_rownames("Contig") 
colnames(reads) <- gsub("_sort.Read.Count", "", colnames(reads))
reads <- reads %>% select(all_of(samps))

dds <- DESeqDataSetFromMatrix(countData = reads,
                              colData = design_phage,
                              design= ~ grp)
dds <- DESeq(dds)
res <- results(dds)
resultsNames(dds)

dds_counts <- counts(dds, normalized=TRUE)
rownames(dds_counts) <- rownames(res)
dds_votu_tax <- merge(dds_counts, tax, by.x="row.names", by.y="Contig")
write.csv(dds_votu_tax, file = "dds_votu_tax.csv")
saveRDS(dds, file = "dds_votu_tax.rds")

# Load existing DDS
dds <- readRDS("dds_votu_tax.rds")

# P vs. no P
res <- results(dds, tidy=TRUE, contrast=c("grp", "P_1", "None_0")) %>%
  arrange(padj, pvalue) 
colnames(res)[1] <- "Contig"
res <- res %>% tibble() %>% left_join(tax, by = "Contig")

EnhancedVolcano(res,
    lab = res$taxonomy,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 10e-10,
    FCcutoff = 1,
    pointSize = 1.0,
    labSize = 1.5,
    shape = 1,
    title = "With phosphate amendment T1 vs. T0")

sig_taxa <- res %>% filter(padj < 10e-6) 
sig_taxa_down <- sig_taxa %>% filter(log2FoldChange < 0) # Narna and Ourli
sig_taxa_up <- sig_taxa %>% filter(log2FoldChange > 0) # mostly Narna and Levi
ggplot(sig_taxa_up, aes(x = taxonomy, y = log2FoldChange)) + geom_boxplot(aes(fill = taxonomy)) + 
  theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_brewer(palette = "Set3")
ggsave("vOTUs_deseq_up_with_P_p10-6.pdf", height = 9, width = 8.5)
n_per_tax <- sig_taxa_up %>% group_by(taxonomy) %>% tally()
# Save data
votus <- reads %>% rownames_to_column("Contig") %>% filter(Contig %in% sig_taxa$Contig) %>% left_join(tax, by = "Contig") %>% left_join(hosts, by = "Contig")
write.table(votus, "table_S8_votus.tsv", sep = ",", row.names = F)

# Plot heatmap of vOTUs that increased with P amendment between T1 and T0
Pvotus <- wetup %>% rownames_to_column("Contig") %>% filter(Contig %in% votus$Contig) %>% pivot_longer(cols = !Contig, names_to = "sample", values_to = "TPM") %>% mutate(grp = gsub("\\.[A-C]$", "", sample)) %>% mutate(grp = gsub("\\.R\\.H2.*O", "", grp)) %>% group_by(Contig, grp) %>% summarise(TPM = sum(TPM)) %>% pivot_wider(names_from = grp, values_from = TPM) %>% left_join(hosts) %>% left_join(tax) %>% column_to_rownames("Contig")

tpm <- Pvotus %>% select(starts_with("T", ignore.case = FALSE))
ann <- Pvotus %>% select(Order, host)
ann_colors = list(
    host = c(Bacteria = "#6fa8dc", Fungi = "#ea9999"),
    Order = c(Timlovirales = "#FFFF99", Levivirales = "#6A3D9A", Lenarviricota = "#CAB2D6", Tolivirales = "#B2DF8A", Ourlivirales = "#FF7F00", Pisuviricota = "cyan", Cryppavirales = "#FDBF6F", Wolframvirales = "#B15928")
)

pheatmap(tpm, color = colorRampPalette(brewer.pal(n = 9, name = "BrBG"))(100), scale = "row", cluster_rows = TRUE, cluster_cols = FALSE, legend = TRUE, annotation_row = ann, annotation_colors = ann_colors, show_rownames = FALSE, filename = "Pvotus_up_heatmap.pdf")

# T1 unamended vs. T0
res <- results(dds, tidy=TRUE, contrast=c("grp", "None_1", "None_0")) %>%
  arrange(padj, pvalue) 
colnames(res)[1] <- "Contig"
res <- res %>% tibble() %>% left_join(tax, by = "Contig")

EnhancedVolcano(res,
    lab = res$taxonomy,
    x = 'log2FoldChange',
    y = 'pvalue',
    pCutoff = 10e-10,
    FCcutoff = 1,
    pointSize = 1.0,
    labSize = 1.5,
    shape = 1,
    title = "With phosphate amendment T1 vs. T0")

sig_taxa <- res %>% filter(padj < 10e-6) 
sig_taxa_down <- sig_taxa %>% filter(log2FoldChange < 0) # None
sig_taxa_up <- sig_taxa %>% filter(log2FoldChange > 0) # Narna and Levi
ggplot(sig_taxa_up, aes(x = taxonomy, y = log2FoldChange)) + geom_boxplot(aes(fill = taxonomy)) + 
  theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_brewer(palette = "Set3")
ggsave("vOTUs_deseq_up_without_P_p10-6.pdf", height = 9, width = 8.5)
n_per_tax <- sig_taxa_up %>% group_by(taxonomy) %>% tally()
# Save data
votus_up <- reads %>% rownames_to_column("Contig") %>% filter(Contig %in% sig_taxa_up$Contig) %>% left_join(tax, by = "Contig") %>% left_join(hosts, by = "Contig")
votus_down <- reads %>% rownames_to_column("Contig") %>% filter(Contig %in% sig_taxa_down$Contig) %>% left_join(tax, by = "Contig") %>% left_join(hosts, by = "Contig")
write.table(votus_up, "table_S9_votus_up.tsv", sep = ",", row.names = F)
write.table(votus_up, "table_S10_votus_down.tsv", sep = ",", row.names = F)

sig_taxa <- wetup %>% rownames_to_column("Contig") %>% filter(Contig %in% sig_taxa_down$Contig) %>% pivot_longer(cols = !Contig, names_to = "sample", values_to = "TPM") %>% mutate(grp = gsub("\\.[A-C]$", "", sample)) %>% mutate(grp = gsub("\\.R\\.H2.*O", "", grp)) %>% group_by(Contig, grp) %>% summarise(TPM = sum(TPM)) %>% pivot_wider(names_from = grp, values_from = TPM) %>% left_join(hosts) %>% left_join(tax) %>% column_to_rownames("Contig")
tpm <- sig_taxa %>% select(starts_with("T", ignore.case = FALSE))
ann <- sig_taxa %>% select(Order, host)
ann_colors = list(
    host = c(Bacteria = "#6fa8dc", Fungi = "#ea9999"),
    Order = c(Wolframvirales = "#FFFF99", Ourlivirales = "#FF7F00")
)

pheatmap(tpm, color = colorRampPalette(brewer.pal(n = 9, name = "BrBG"))(100), scale = "row", cluster_rows = TRUE, cluster_cols = FALSE, legend = TRUE, annotation_row = ann, annotation_colors = ann_colors, show_rownames = FALSE)
```
