---
title: "RNA viruses"
author: "Ella Sieradzki"
date: "2022-12-27"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
library("tidyverse")
library("data.table")
library("ggplot2")
library("vegan")
library("RColorBrewer")
library("ape")
library("ggpubr")
library("ggVennDiagram")
library("ggpmisc")
library("abdiv")
library("chemodiv")
library("taxonomizr")
#library("phyloseq")
```

## Load data

```{r votu table, echo=FALSE}
votu <- read.delim("coverm_rpkm_breadth50_Gary.tsv", sep="\t", header=T, stringsAsFactors = F)
str(votu)
colnames(votu) <- gsub("_sort\\.RPKM", "", colnames(votu))
votus_rmv <- scan(file = "vOTUs_to_remove_unclassified.txt",what = character())
votu <- votu %>%
  filter(!(Contig %in% votus_rmv))

wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]
# To account for T0 under both treatments (P amended and unamended), replicate the T0 values and assign once to P and once to noP
toadd <- wetup[, 1:3]
colnames(toadd) <- gsub("A$", "P.A", colnames(toadd))
colnames(toadd) <- gsub("B$", "P.B", colnames(toadd))
colnames(toadd) <- gsub("C$", "P.C", colnames(toadd))
wetup <- data.frame(cbind(toadd, wetup))

wetup <- t(wetup)

design <- read.delim("mtx_design_Gary.txt")
```

## Create Permanova

```{r}
adonis2(wetup~time*Phos*isotope, data=design)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = wetup ~ time * Phos * isotope, data = design)
#                   Df SumOfSqs      R2      F Pr(>F)
# time               1   0.9070 0.11800 6.5940  0.001 ***
# Phos               1   0.8090 0.10525 5.8815  0.001 ***
# isotope            1   0.3980 0.05178 2.8932  0.003 **
# time:Phos          1   0.2452 0.03190 1.7827  0.053 .
# time:isotope       1   0.3709 0.04826 2.6966  0.003 **
# Phos:isotope       1   0.1576 0.02050 1.1456  0.280
# time:Phos:isotope  1   0.1219 0.01586 0.8860  0.492
# Residual          34   4.6769 0.60845
# Total             41   7.6865 1.00000
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

# PERMANOVA without T0
design_no_T0 <- design %>% filter(time>0)
wetup_no_T0 <- wetup[-(1:6),]
adonis2(wetup_no_T0~time*Phos*isotope, data=design_no_T0)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = wetup_no_T0 ~ time * Phos * isotope, data = design_no_T0)
#                   Df SumOfSqs      R2      F Pr(>F)    
# time               1   0.3089 0.05404 2.5536  0.005 ** 
# Phos               1   0.9439 0.16510 7.8021  0.001 ***
# isotope            1   0.4182 0.07315 3.4568  0.001 ***
# time:Phos          1   0.2145 0.03752 1.7732  0.040 *  
# time:isotope       1   0.1468 0.02568 1.2138  0.194    
# Phos:isotope       1   0.1788 0.03127 1.4777  0.098 .  
# time:Phos:isotope  1   0.1186 0.02074 0.9802  0.406    
# Residual          28   3.3873 0.59250                  
# Total             35   5.7170 1.00000                  
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

# Square root transformation
wetup_sqrt <- sqrt(wetup)
wetup_dist <- vegdist(wetup_sqrt, method="bray", na.rm = T)
wetup_cols <- factor(design$cosm)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_")) %>%
  mutate(ellps = as.factor(gsub("T[1-3]_", "", cond)))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample, group=ellps)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  stat_ellipse() +
  theme_bw()
ggsave("Gary_RNA_viruses_PCoA_wetup_sqrt.pdf", width=10)
# Recalculate without T0 to see if later temporal patterns pop up
wetup_sqrt <- wetup_sqrt[!(rownames(wetup_sqrt) %like% "T0"),]
wetup_dist <- vegdist(wetup_sqrt, method="bray", na.rm = T)
pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  #geom_text(size=1) +
  theme_bw()
# Nah, no epiphany here 
```
```{r}
# vOTU sharing between replicates, supplemental figure
wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]
wetup_df[wetup_df > 0] <- 1 # convert coverage to presence/absence

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "Dry.soil.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
fig0 <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H218O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT1_18O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H218O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT1_18O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H2O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT1_16O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T1.R.H2O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT1_16O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H218O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT2_18O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H218O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT2_18O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H2O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT2_16O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T2.R.H2O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT2_16O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H218O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT3_18O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H218O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT3_18O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H2O.P.*")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT3_16O_P <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")
sums <- rowSums(wetup_df[, which(colnames(wetup_df) %like% "T3.R.H2O.[A-C]")])
sums2 <- c(length(sums[sums==1]), length(sums[sums==2]), length(sums[sums==3]))
sums3 <- data.frame(t(rbind(c("1", "2", "3"), sums2)))
figT3_16O_noP <- ggplot(data = sums3, aes(x = V1, y = sums2, fill = V1)) + geom_col() + scale_fill_brewer(palette = "Purples") + theme_classic() + xlab("Number of replicates") + ylab("Number of vOTUs")

ggarrange(fig0, 
          ggarrange(figT1_18O_P, figT1_18O_noP, figT1_16O_P, figT1_16O_noP, nrow=1, ncol=4, common.legend = T),
          ggarrange(figT2_18O_P, figT2_18O_noP, figT2_16O_P, figT2_16O_noP, nrow=1, ncol=4, common.legend = T),
          ggarrange(figT3_18O_P, figT3_18O_noP, figT3_16O_P, figT3_16O_noP, nrow=1, ncol=4, common.legend = T),
          nrow = 4)
ggsave("fig_S1_replication_hist.pdf", width=8, height=10)

# Number of samples each vOTU appears in histogram
wetup_df <- wetup_df %>% select(-matches("T0.*P.*"))
sums <- rowSums(wetup_df)

hist(sums, breaks=39, xlab = "Number of microcosms", ylab = "Number of vOTUs", main = "")
```

```{r}
# Taxonomy distribution
wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]

tax <- read.delim("Gary_RdRp_score70_verified_RdRp.tax", sep="\t", header=F, stringsAsFactors = F)
colnames(tax) <- c("RdRp_ID", "taxonomy", "bitscore", "Contig")
ictv <- read.delim("ICTV_RNA_viruses.txt", sep="\t", header=T, stringsAsFactors = F)
tax$Class <- tax$taxonomy

getOrder <- function(taxname) {
  if ((taxname %like% "ota$") | (taxname %like% "tes$") | (taxname %like% "les$")) return(taxname) # if it's already a phylum or class name keep it as is
  if (taxname %in% c("Zhaovirus", "Yanvirus", "Permutotetraviridae", "Ophioviridae")) return("Unassigned") # These exceptions don't have a class or phylum
  if (taxname %like% "dae$") return(unique(ictv$Order[ictv$Family == taxname]))
  else return(unique(ictv$Order[ictv$Genus == taxname]))
}
getClass <- function(taxname) {
  if ((taxname %like% "ota$") | (taxname %like% "tes$")) return(taxname) # if it's already a phylum or class name keep it as is
  if (taxname == "Levivirales") return("Leviviricetes") # Levivirales don't exist anymore so I had to define this manually
  if (taxname == "Ophioviridae") return("Milneviricetes") # Ophioviridae is a new family so I had to define this manually
  if (taxname %in% c("Zhaovirus", "Yanvirus", "Permutotetraviridae")) return("Unassigned") # These exceptions don't have a class or phylum
  if (taxname %like% "les$") return(unique(ictv$Class[ictv$Order == taxname]))
  if (taxname %like% "dae$") return(unique(ictv$Class[ictv$Family == taxname]))
  else return(unique(ictv$Class[ictv$Genus == taxname]))
}

tax$Class <- unlist(sapply(tax$taxonomy, getClass))
tax$Order <- unlist(sapply(tax$taxonomy, getOrder))

view(tax[which(duplicated(tax$Contig)),]) # Empty = No duplicates
# If not empty run this
remrows <- c()
for (r in which(duplicated(tax$Contig))) {
  ctg <- tax$Contig[r]
  bit <- min(tax$bitscore[tax$Contig == ctg])
  remrows <- c(remrows, which(tax$Contig == ctg & tax$bitscore == bit))
}
tax <- tax[-remrows,]

write.csv(tax, "Order_level_tax.tsv", row.names = FALSE)
# To load the existing table
tax <- read.csv("Order_level_tax.tsv")

tax_vOTU <- tax %>%
  select(Contig, Class, Order, taxonomy) %>%
  unique()

wetup_tax <- wetup %>% rownames_to_column("Contig") %>% left_join(tax_vOTU, by="Contig")

wetup_cols <- which(colnames(wetup_tax) %like% "^T")
wetup <- wetup_tax %>% select(Contig, taxonomy, starts_with("T")) %>%
  filter(rowSums(wetup_tax[,wetup_cols]) > 0, na.rm = TRUE) %>% unique() # 3330 vOTUs

# Based on the PCoA I suspect the names of T3 P rep C 16O and 18O were accidentally swapped on the tubes. 
# Swapping them here
#wetup <- wetup %>% rename(T3.R.H218O.P.C = T3.R.H2O.P.C, T3.R.H2O.P.C = T3.R.H218O.P.C)

wetup_pvt <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Class, Order, taxonomy), names_to="variable", values_to="value") %>%
  mutate(pvt = ifelse(value > 0, 1, 0)) %>%
  group_by(Class, variable) %>%
  summarize_at("pvt", sum) %>%
  mutate(variable = as.character(variable))
top11 <- wetup_pvt %>% group_by(Class) %>% summarize_at("pvt", sum)
top11 <- head(top11$Class[order(-top11$pvt)], 10)
wetup_tax_plot <- wetup_pvt %>%
  mutate(Class = ifelse(Class %in% top11, Class, "Other")) %>%
  mutate(Class = factor(Class))
top11 <- c(top11, "Other")
wetup_tax_plot$Class <- fct_relevel(wetup_tax_plot$Class, top11)
# Bar chart
ggplot(data = wetup_tax_plot, aes(x = variable, y = pvt, fill=Class)) + 
  geom_bar(stat = "identity", position = "stack") + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("Sample") + ylab("Number of vOTUs present")
ggsave("wetup_richness_tax_bar_class.pdf")
# Boxplot
ggplot(data = wetup_tax_plot, aes(x = Class, y = pvt, fill=Class)) + 
  geom_boxplot() + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("taxonomy") + ylab("Number of vOTUs per sample")
ggsave("wetup_richness_tax_box_class.pdf")
# Relative abundance stacked barchart
for_durna <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Class, Order, taxonomy), names_to="variable", values_to="rpkm") %>%
  group_by(Order, variable) %>%
  summarize_at("rpkm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = rpkm*100/sum(rpkm))
wetup_pct <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Class, Order, taxonomy), names_to="variable", values_to="rpkm") %>%
  group_by(Class, variable) %>%
  summarize_at("rpkm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = rpkm*100/sum(rpkm))

wetup_tax_plot <- wetup_pct %>% 
  mutate(Class = ifelse(Class %in% top11, Class, "Other")) %>%
  mutate(Class = factor(Class))
wetup_tax_plot$Class <- fct_relevel(wetup_tax_plot$Class, top11)

# Most phages are Leviviricetes or Durnavirales
levi <- wetup_tax_plot %>% filter(Class %like% "Levivi") # Save Leviviricetes for later script
durna <- for_durna %>% filter(Order %like% "Durnavi") # Save Durnavirales phages for later (see Neri 2022)

ggplot(data = wetup_tax_plot, aes(x = variable, y = pct, fill=Class)) + 
  geom_bar(stat = "identity", position = "fill") + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("Sample") + ylab("Relative abundance")
ggsave("wetup_relabun_tax_bar_class.pdf")
# Boxplot
ggplot(data = wetup_tax_plot, aes(x = Class, y = pct, fill=Class)) + 
  geom_boxplot() + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("taxonomy") + ylab("Relative abundance per sample")
ggsave("wetup_relabun_tax_box_class.pdf")

# Is there a correlation between richness and abundance?
temp <- wetup_pct %>% left_join(wetup_pvt, by=c("Class", "variable"))
temp$Class[!(temp$Class %in% top11)] <- "Other"
temp$Class <- fct_relevel(temp$Class, top11)
ggplot(data=temp, aes(x=pvt, y=pct)) + geom_point(aes(fill=Class), colour="black",pch=21, size=2) + 
  scale_fill_brewer(palette = "Paired", direction = -1) + 
  theme_classic() +
  stat_poly_line(color="black") +
  stat_poly_eq(use_label(c("eq", "adj.R2", "f", "p", "n"))) +
  xlab("Number of unique vOTUs from taxonomic group")+ ylab("Relative abundance of vOTUs from taxonomic group")
ggsave("wetup_richness_vs_abundance_class.pdf")
```
```{r}
# Special vOTUs
dry_cols <- which(colnames(wetup_tax) %like% "T0")
p_cols <- which(colnames(wetup_tax) %like% "P")
nop_cols <- c(which(colnames(wetup_tax) %like% "T[1-3].*O.[A-C]"))

# Only in dry soil
tmp1 <- wetup_tax[rowSums(wetup_tax[,dry_cols]) > 0 & rowSums(wetup_tax[,c(p_cols, nop_cols)]) == 0,]
nrow(tmp1) #42
# Only in P amended soil
tmp2 <- wetup_tax[rowSums(wetup_tax[,p_cols]) > 0 & rowSums(wetup_tax[,nop_cols]) == 0,]
nrow(tmp2) #525
# Only in unamended soil
tmp3 <- wetup_tax[rowSums(wetup_tax[,p_cols]) == 0 & rowSums(wetup_tax[,nop_cols]) > 0,]
nrow(tmp3) #683

tmp_long <- tmp1 %>% select(starts_with("T0"), "Order", "Class") %>% pivot_longer(cols = starts_with("T0"), names_to = "sample") %>% group_by(Class) %>% summarise(total = sum(value)) %>% arrange(desc(total))
tmp_long$Class <- fct_relevel(tmp_long$Class, tmp_long$Class)
p1 <- ggplot(tmp_long, aes(x = 1, y=total)) + geom_bar(stat = "identity", aes(fill = Class), position = "stack") + theme_bw() + scale_fill_brewer(palette = "Paired")

tmp_long <- tmp2 %>% select(contains("P"), "Order", "Class") %>% pivot_longer(cols = contains("P"), names_to = "sample") %>% group_by(Class) %>% summarise(total = sum(value)) %>% arrange(desc(total))
top11 <- tmp_long$Class[1:11]
tmp_long$Class[!(tmp_long$Class %in% top11)] <- "Other"
tmp_long$Class <- fct_relevel(tmp_long$Class, c(top11, "Other"))
p2 <- ggplot(tmp_long, aes(x = 1, y=total)) + geom_bar(stat = "identity", aes(fill = Class), position = "stack") + theme_bw() + scale_fill_brewer(palette = "Paired")

tmp_long <- tmp3 %>% select(matches("T[1-3].*O.[A-C]"), "Order", "Class") %>% pivot_longer(cols =starts_with("T"), names_to = "sample") %>% group_by(Class) %>% summarise(total = sum(value)) %>% arrange(desc(total))
top11 <- tmp_long$Class[1:11]
tmp_long$Class[!(tmp_long$Class %in% top11)] <- "Other"
tmp_long$Class <- fct_relevel(tmp_long$Class, c(top11, "Other"))
p3 <- ggplot(tmp_long, aes(x = 1, y=total)) + geom_bar(stat = "identity", aes(fill = Class), position = "stack") + theme_bw() + scale_fill_brewer(palette = "Paired")


```

```{r}
# Rerun Permanova on Leviviricetes only
temp <- wetup_tax %>% filter(Class %like% "Levivi") %>%
  column_to_rownames("Contig") %>%
  select(starts_with("T", ignore.case = FALSE))
toadd <- temp[, 1:3]
colnames(toadd) <- gsub("A$", "P.A", colnames(toadd))
colnames(toadd) <- gsub("B$", "P.B", colnames(toadd))
colnames(toadd) <- gsub("C$", "P.C", colnames(toadd))
temp <- data.frame(cbind(toadd, temp))
tempt <- data.frame(t(temp))
adonis2(tempt~time*Phos, data=design)

# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = tempt ~ time * Phos, data = design_temp)
#           Df SumOfSqs      R2       F Pr(>F)    
# time       1   0.9197 0.12839  7.2891  0.001 ***
# Phos       1   1.2675 0.17694 10.0457  0.001 ***
# time:Phos  1   0.1815 0.02534  1.4388  0.157    
# Residual  38   4.7947 0.66932                   
# Total     41   7.1634 1.00000                   
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


```
```{r}
# Histogram of number of samples per vOTU
temp <- wetup_tax %>% select(-Order) %>%
  pivot_longer(cols = !c(Contig, taxonomy, Class), names_to = "sample", values_to = "count") %>%
  mutate(count = case_when(
    count > 0 ~ 1,
    TRUE ~ 0
  )) %>%
  group_by(Contig, Class) %>% summarise_at("count", sum) %>%
  mutate(Class = case_when(
    Class %in% top11 ~ Class,
    TRUE ~ "Other"
   )) #%>%
  # mutate(Class = factor(Class, levels=c(top11, "Other")))

ggplot(data=temp, aes(fill=Class, x = count)) + geom_histogram(stat="bin", binwidth = 1) + 
  scale_fill_brewer(palette="Paired", direction=-1) + theme_classic() + xlab("Number of samples") + ylab("Number of vOTUs")
ggsave("vOTUs_shared_hist.pdf", width=9)

# Taxonomic profiles of vOTUs shared over X samples
temp_dis <- temp %>%
  group_by(Class, count) %>%
  tally %>%
  arrange(count, desc(n))  %>%
  mutate(rank = 1)
for (c in 1:39) {
  temp_dis$rank[temp_dis$count == c] <- 1:nrow(temp_dis[temp_dis$count == c,])
}
  
ggplot(data=temp_dis, aes(x = Class, y = rank, fill = Class)) + 
  theme_classic() + geom_boxplot() + xlab("Phylum") +  ylab("Rank") + 
  scale_fill_brewer(palette = "Paired", direction = -1) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ggsave("sup_fig_phylum_mean_rankabun.pdf")  
```

```{r}
# Compare phylum level distributions
tax <- read.csv("Phylum_level_tax.tsv")

tax_vOTU <- tax %>%
  select(Contig, Phylum) %>%
  unique()

wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]
wetup <- t(wetup)

wetup_tax <- t(wetup) %>% data.frame %>% rownames_to_column("Contig") %>% left_join(tax_vOTU, by="Contig")
wetup_tax$taxonomy <- wetup_tax$Phylum

getPhylum <- function(taxname) {
  if (taxname %like% "ota$") return(taxname) # if it's already a phylum or class name keep it as is
  if (taxname == "Levivirales") return("Lenarviricota") # Levivirales don't exist anymore so I had to define this manually
  if (taxname == "Ophioviridae") return("Negarnaviricota") # Ophioviridae is a new family so I had to define this manually
  if (taxname %in% c("Zhaovirus", "Yanvirus", "Permutotetraviridae")) return("Unassigned") # These exceptions don't have a class or phylum
  if (taxname %like% "tes$") return(unique(ictv$Phylum[ictv$Class == taxname]))
  if (taxname %like% "les$") return(unique(ictv$Phylum[ictv$Order == taxname]))
  if (taxname %like% "dae$") return(unique(ictv$Phylum[ictv$Family == taxname]))
  else return(taxname)
  #else return(unique(ictv$Phylum[ictv$Genus == taxname]))
}

wetup_tax$Phylum <- unlist(sapply(wetup_tax$taxonomy, getPhylum))

# Convert numbers to percentages
wetup_pct <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Phylum, taxonomy), names_to="variable", values_to="rpkm") %>%
  group_by(Phylum, variable) %>%
  summarize_at("rpkm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = rpkm*100/sum(rpkm)) %>%
  mutate(Phylum = factor(Phylum))

# Plot mean abundance by phylum
wetup_pct <- wetup_tax %>% 
  remove_rownames() %>%
  column_to_rownames("Contig") %>%
  pivot_longer(cols=-c(Phylum, taxonomy), names_to="variable", values_to="rpkm") %>%
  group_by(Phylum, variable) %>%
  summarize_at("rpkm", sum) %>%
  mutate(variable = as.character(variable)) %>%
  group_by(variable) %>%
  mutate(pct = rpkm*100/sum(rpkm))

ggplot(data = wetup_pct, aes(x = Phylum, y = pct, fill=Phylum)) + 
  geom_boxplot() + 
  scale_fill_brewer(palette = "Pastel2", direction = -1) + 
  theme_classic() + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  xlab("taxonomy") + ylab("Relative abundance per sample")
ggsave("wetup_relabun_tax_box_phylum.pdf")

# Plot temporal abundance by phylum
# Mean and SE by Phylum
mean_ses <- wetup_pct %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", variable))) %>%
  mutate(amend = case_when(
    variable %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  group_by(Phylum, time) %>%
  mutate(pct_mean = mean(pct), pct_sd = sd(pct)) %>%
  mutate(sd_div_mean = pct_sd/pct_mean) %>%
  arrange(desc(pct_mean))

ggplot(mean_ses, aes(x = pct_mean, y = sd_div_mean)) + geom_point() + scale_x_continuous(trans='log2') +
  geom_smooth(method = lm) + stat_regline_equation(formula = y ~ x) + stat_cor(label.y=-0.1) + theme_classic()
# sd becomes a smaller % of the mean as the mean grows - not a great correlation

# Temporal dynamics by phylum for viruses
df <- wetup_tax %>% left_join(hosts, by="Contig")
df <- df %>% column_to_rownames("Contig") %>% pivot_longer(cols=starts_with("T", ignore.case = FALSE), names_to = "sample", values_to = "rpkm") %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", sample))) %>%
  mutate(amend = case_when(
    sample %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  mutate(replicate = gsub("T.*\\.", "", sample))
wP <- df %>% filter(amend == "P" | time == 0) %>% group_by(Phylum, time, amend, replicate) %>% summarise(rpkm_sum = sum(rpkm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(rpkm_mean = mean(rpkm_sum), sd = sd(rpkm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))
woP <- df %>% filter(amend == "noP") %>% group_by(Phylum, time, amend, replicate) %>% summarise(rpkm_sum = sum(rpkm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(rpkm_mean = mean(rpkm_sum), sd = sd(rpkm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))

pal <- rev(brewer.pal(6, "Blues"))
PP <- ggplot(data=wP, aes(x=time, y=rpkm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
PnoP <- ggplot(data=woP, aes(x=time, y=rpkm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("rpkm_sum_byphylum_line.pdf", width = 8, height = 4)
rpkm_wohost <- sum(df$rpkm)

l_R <- list()
l_p <- list()
l_perm <- list()
phyla <- c("Duplornaviricota", "Pisuviricota", "Lenarviricota", "Kitrinoviricota", "Negarnaviricota")
for (i in 1:5) {
  p <-  phyla[i]
  temp <- wetup_tax %>% filter(Phylum ==  p) %>% select(!c("Phylum", "taxonomy")) %>% column_to_rownames("Contig") %>% t %>% as.data.frame()
  temp <- temp[rowSums(temp) > 0,]
  wetup_tax_dist <- vegdist(temp, method = "bray")
  temp_design <- design[design$Sample %in% rownames(temp),]
  templ <- adonis2(wetup_tax_dist~time*Phos, data=temp_design)
  l_perm[[i]] <- templ
  l_R[[i]] <- templ$R2
  l_p[[i]] <- templ$`Pr(>F)`
}
# correct p values for multiple comparisons (3 per class * 10 classes)
getsig <- function (x) {p.adjust(x, method="fdr", n=30)}
l_p <- lapply(l_p, getsig)
getsig <- function (x) {which(x <= 0.05)}
sigp <- lapply(l_p, getsig) # get indices of significant effects per phylum
sigR <- map2(l_R, sigp, `[`) # get R^2 values of these effects
# dsDNA: Duplorna unaffected, Pisu affected by all. ssRNA+: Lenar by time and P, Kitrino by all. ssRNA-: Negarna by time.
```
```{r}
# Host prediction
hosts <- read.delim(file = "Trees/votus_w_host_assignment_uniq.txt", sep=" ", header=F)
colnames(hosts) <- c("Contig", "host")
for_pie <- hosts %>% group_by(host) %>% tally() %>% arrange(desc(n)) %>% mutate(host = factor(host, levels = c("Bacteria", "Fungi", "Invertebrates", "Plants", "Eukaryotes", "Mammals", "Sauria")))
# for_pie_n <- for_pie$n
# names(for_pie_n) <- rownames(for_pie)
# pie(for_pie_n, col = brewer.pal(length(for_pie_n), "Pastel2"))
ggplot(data=for_pie, aes(x = host, y=n, fill=host)) + geom_col(position = "stack") + theme_classic() + 
  scale_fill_brewer(palette = "Pastel1")

# Temporal dynamics by phylum for viruses with an assigned host
wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]
wetup <- t(wetup)

df <- wetup_tax %>% left_join(hosts, by="Contig") %>% filter(!is.na(host))
df <- df %>% column_to_rownames("Contig") %>% pivot_longer(cols=starts_with("T", ignore.case = FALSE), names_to = "sample", values_to = "rpkm") %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", sample))) %>%
  mutate(amend = case_when(
    sample %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  mutate(replicate = gsub("T.*\\.", "", sample))
rpkm_hosts <- sum(df$rpkm)
wP <- df %>% filter(amend == "P" | time == 0) %>% group_by(Phylum, time, amend, replicate) %>% summarise(rpkm_sum = sum(rpkm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(rpkm_mean = mean(rpkm_sum), sd = sd(rpkm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))
woP <- df %>% filter(amend == "noP") %>% group_by(Phylum, time, amend, replicate) %>% summarise(rpkm_sum = sum(rpkm)) %>% 
  group_by(Phylum, time, amend) %>% summarise(rpkm_mean = mean(rpkm_sum), sd = sd(rpkm_sum)) %>%
  mutate(Phylum = factor(Phylum, levels = c("Lenarviricota", "Kitrinoviricota", "Negarnaviricota", "Duplornaviricota", "Pisuviricota")))

pal <- rev(brewer.pal(6, "Blues"))
PP <- ggplot(data=wP, aes(x=time, y=rpkm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
PnoP <- ggplot(data=woP, aes(x=time, y=rpkm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") + ylim(c(0, 600000)) +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("rpkm_sum_byphylum_line.pdf", width = 8, height = 4)

pal <- rev(brewer.pal(5, "Blues"))
temp_wP <- wP %>% filter(!(Phylum %like% "Lenar"))
temp_woP <- woP %>% filter(!(Phylum %like% "Lenar"))
PP <- ggplot(data=temp_wP, aes(x=time, y=rpkm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
PnoP <- ggplot(data=temp_woP, aes(x=time, y=rpkm_mean, colour=Phylum, group=Phylum)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("rpkm_sum_byhost_nolenar_line.pdf", width = 8, height = 4)

# Temporal dynamics by host
df <- wetup_tax %>% left_join(hosts, by="Contig") %>% filter(!is.na(host))
df <- df %>% column_to_rownames("Contig") %>% pivot_longer(cols=starts_with("T", ignore.case = FALSE), names_to = "sample", values_to = "rpkm") %>%
  mutate(time = gsub("T", "", gsub("\\..*", "", sample))) %>%
  mutate(amend = case_when(
    sample %like% "P" ~ "P",
    .default = "noP"
  )) %>%
  mutate(replicate = gsub("T.*\\.", "", sample))
wP <- df %>% filter(amend == "P" | time == 0) %>% group_by(host, time, amend, replicate) %>% summarise(rpkm_sum = sum(rpkm)) %>% 
  group_by(host, time, amend) %>% summarise(rpkm_mean = mean(rpkm_sum), sd = sd(rpkm_sum))
woP <- df %>% filter(amend == "noP") %>% group_by(host, time, amend, replicate) %>% summarise(rpkm_sum = sum(rpkm)) %>% 
  group_by(host, time, amend) %>% summarise(rpkm_mean = mean(rpkm_sum), sd = sd(rpkm_sum))

pal <- rev(brewer.pal(10, "Paired"))
HP <- ggplot(data=wP, aes(x=time, y=rpkm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
HnoP <- ggplot(data=woP, aes(x=time, y=rpkm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") + ylim(c(0, 600000)) +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
ggarrange(PP, PnoP, HP, HnoP, nrow=2, ncol=2, common.legend = TRUE)
ggsave("rpkm_sum_byhost_line.pdf", width = 8, height = 4)

pal <- rev(brewer.pal(9, "Paired"))
temp_wP <- wP %>% filter(!(host %like% "Bacteria"))
temp_woP <- woP %>% filter(!(host %like% "Bacteria"))
PP <- ggplot(data=temp_wP, aes(x=time, y=rpkm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("P amended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
PnoP <- ggplot(data=temp_woP, aes(x=time, y=rpkm_mean, colour=host, group=host)) +
  geom_line() + scale_colour_manual(values = pal) +
  theme_bw() + ggtitle("Unamended") +
  xlab("Time (weeks)") + ylab("Cummulative RPKM") + 
  geom_errorbar(aes(ymin=rpkm_mean-sd, ymax=rpkm_mean+sd), width=0.1)
ggarrange(PP, PnoP, nrow=1, ncol=2, common.legend = TRUE)
ggsave("rpkm_sum_byhost_nophages_line.pdf", width = 8, height = 4)

# Looks like there's high similarity between phylum level dynamics and dynamics of viruses with known hosts:
# Lenarviricota and phages
# Kitrinoviricota and fungal viruses
sum(df$rpkm[(df$host=="Bacteria")]) / sum(df$rpkm) # 87% of rpkm maps to phages
sum(df$rpkm[(df$host=="Bacteria") & (df$Phylum=="Lenarviricota")]) / sum(df$rpkm) # 85% of rpkm maps to Lenar phages
sum(df$rpkm[(df$host=="Fungi")]) / sum(df$rpkm) # 11.4% of rpkm maps to fungal viruses
sum(df$rpkm[(df$host=="Fungi") & (df$Phylum=="Kitrinoviricota")]) / sum(df$rpkm) # 8% of rpkm maps to Kitrino fungal viruses

# The rpkm of viruses with hosts is 43% of rpkm of total viruses with assigned phylum
rpkm_hosts / rpkm_wohost # 0.4277

df2 <- df %>% select(host, Phylum, rpkm) %>% 
  mutate(rpkm = round(rpkm, digits = 0)) %>%
  pivot_wider(names_from = host, values_from = rpkm, values_fn = sum)
df2[is.na(df2)] <- 0
```
```{r}
# Plot median abundance over time per phylum to add error bars
for (p in names(cols)) {
temp <- wetup_P_pers_melt_16 %>% filter(Phylum %like% p)
titl <- paste0(p, "median abundance with P addition 16O-water")
p16_P <- ggplot(temp, aes(x=time, y=RPKM, group=Phylum, color=Phylum)) + geom_line() + theme_bw() +
  scale_color_manual(values = cols) + ylab("log10 RPKM") + 
  ggtitle(titl) + xlab("Time (weeks)") + 
  theme(legend.position = "none") +
  geom_point() +
  geom_errorbar(aes(ymin=RPKM-SD, ymax=RPKM+SD), width=.2,
                 position=position_dodge(0.1))
temp <- wetup_P_pers_melt_18 %>% filter(Phylum %like% p)
titl <- paste0(p, "median abundance with P addition 18O-water")
p18_P <- ggplot(temp, aes(x=time, y=RPKM, group=Phylum, color=Phylum)) + geom_line() + theme_bw() +
  scale_color_manual(values = cols) + ylab("log10 RPKM") + 
  ggtitle(titl) + xlab("Time (weeks)") + 
  theme(legend.position = "none") +
  geom_point() +
  geom_errorbar(aes(ymin=RPKM-SD, ymax=RPKM+SD), width=.2,
                 position=position_dodge(0.1))
temp <- wetup_NoP_pers_melt_16 %>% filter(Phylum %like% p)
titl <- paste0(p, "median abundance without P addition 16O-water")
p16_NoP <- ggplot(temp, aes(x=time, y=RPKM, group=Phylum, color=Phylum)) + geom_line() + theme_bw() + 
  scale_color_manual(values = cols) + ylab("log10 RPKM") + 
  ggtitle(titl) + xlab("Time (weeks)") + 
  theme(legend.position = "none")  +
  geom_point() +
  geom_errorbar(aes(ymin=RPKM-SD, ymax=RPKM+SD), width=.2,
                 position=position_dodge(0.1))
temp <- wetup_NoP_pers_melt_16 %>% filter(Phylum %like% p)
titl <- paste0(p, "median abundance without P addition 18O-water")
p18_NoP <- ggplot(temp, aes(x=time, y=RPKM, group=Phylum, color=Phylum)) + geom_line() + theme_bw() +
  scale_color_manual(values = cols) + ylab("log10 RPKM") + 
  ggtitle(titl) + xlab("Time (weeks)") + 
  theme(legend.position = "none")  +
  geom_point() +
  geom_errorbar(aes(ymin=RPKM-SD, ymax=RPKM+SD), width=.2,
                 position=position_dodge(0.1))

filename <- paste0("wetup_temp_dyn_by_phylum_P_iso_", paste0(p, ".pdf"))
ggarrange(p16_P, p18_P, p16_NoP, p18_NoP, nrow=2, ncol=2)
ggsave(filename)
}
```

```{r}
# Diversity and evenness indices
tax <- read.csv("Phylum_level_tax.tsv")

tax_vOTU <- tax %>%
  select(Contig, Phylum) %>%
  unique()

wetup <- votu %>% column_to_rownames("Contig")
wetup <- wetup[rowSums(wetup) > 0,]

# wetup_shan <- data.frame(sapply(wetup, shannon)) %>% rownames_to_column("Sample") %>%
#   mutate(time = gsub("\\..*", "", gsub("T", "", Sample)))
# colnames(wetup_shan)[2] <- "Shannon_ind"
# summary(aov(Shannon_ind~time, data = wetup_shan)) # p=0.09
# 
# # Shannon evenness doesn't change over time
# ggbarplot(wetup_shan, x="Sample", y="Shannon_ind", fill="time") + scale_fill_brewer(palette="Paired") +
#   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
# ggplot(data=wetup_shan, aes(x = time, y = Shannon_ind, fill = time)) + geom_boxplot() +
#   scale_fill_brewer(palette="Paired") + theme_bw() +
#   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + 
#   xlab("Time (weeks)") + ylab("Shannon evenness index")
# ggsave("wetup_Shannon_evenness_over_time.pdf")

# Diversity of viral communities over time after wetup

# Set palette colors
pal <- rev(brewer.pal(5, "Paired"))
# Shannon-Weaver diversity
H <- data.frame(diversity(t(wetup), "shannon")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(H)[2] <- "Shannon_div"
p_H <- ggplot(H, aes(x=treats, y=Shannon_div, fill=treats)) + 
  geom_boxplot() + 
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) +
  theme_bw() + ylab("Shannon diversity") + theme(axis.title.x=element_blank())
# Simpson index
simp <- data.frame(diversity(t(wetup), "simpson")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(simp)[2] <- "Simpson_div"
p_simp <- ggplot(simp, aes(x=treats, y=Simpson_div, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) + theme(legend.position="none") +
  theme_bw() + ylab("Simpson diversity") + theme(axis.title.x=element_blank())
# Inverse Simpson index
invsimp <- data.frame(diversity(t(wetup), "inv")) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(invsimp)[2] <- "Inv_Simpson_div"
p_invsimp <- ggplot(invsimp, aes(x=treats, y=Inv_Simpson_div, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x=element_blank(), axis.title.x = element_blank()) + theme(legend.position="none") +
  theme_bw() + ylab("Inverse Simpson diversity") + theme(axis.title.x=element_blank())
# Species number
S <- data.frame(specnumber(t(wetup))) %>% rownames_to_column("Sample") %>%
  mutate(treats = gsub("\\.[A-C]$", "", gsub("T[0-3]\\.R\\.", "", Sample)))
colnames(S)[2] <- "Spec_num"
p_S <- ggplot(S, aes(x=treats, y=Spec_num, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Species number") + theme(axis.title.x=element_blank())

# Evenness of viral communities over time after wetup
evenness <- data.frame(cbind(calcDiv(sampleData=t(wetup), type="PielouEven")), S$Sample)
evenness <- data.frame(evenness, S$treats)
colnames(evenness)[2:3] <- c("Sample", "treats")
evenness$Time <- gsub("T", "", gsub("\\..*", "", evenness$Sample))
p_E <- ggplot(evenness, aes(x=treats, y=PielouEven, fill=treats)) + 
  geom_boxplot() +
  scale_fill_manual(values=pal) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), text = element_text(size=12)) + 
  theme(legend.position="none") +
  theme_bw() + ylab("Pielou's evenness") + xlab("Treatment groups")

ggarrange(p_H, p_simp, p_invsimp, p_S, p_E, nrow=5, ncol=1, common.legend = TRUE)
ggsave("diversity_indices.pdf", width=8.5, height=10)

# Statistical analysis and p value adjustment for multiple comparisons
kruskal.test(Shannon_div~treats, data = H) # p=0.007
kruskal.test(Simpson_div~treats, data = simp) # p=0.05
kruskal.test(Inv_Simpson_div~treats, data = invsimp) # p=0.05
kruskal.test(Spec_num~treats, data = S) # p=0.2
kruskal.test(PielouEven~treats, data = evenness) # p=0.002
p.adjust(c(0.0066, 0.048, 0.048, 0.2, 0.0024)) # 0.03, 0.14, 0.14, 0.2, 0.01
```

```{r}
# Plot respiration
resp <- read.delim("respiration.tsv", sep="\t", header=T)
#colnames(resp)
resp <- resp %>% 
  mutate(CO2_ppm = as.numeric(gsub(",", "\\.", CO2_ppm))) %>%
  mutate(time = case_when(
  sample_ID %like% "std" ~ NA,
  sample_ID %like% "blank" ~ 0,
  .default = as.numeric(gsub("W.*", "", gsub("^.*T", "", sample_ID)))
)) %>%
  mutate(amend = case_when(
    sample_ID %like% "P" ~ "P",
    .default = "No P"
  )) %>% 
  select(time, CO2_ppm, amend) %>% pivot_longer(cols = CO2_ppm, values_to = "CO2") %>%
  filter(!is.na(time)) %>%
  group_by(time, amend) %>% summarise(CO2_mean = mean(CO2), CO2_sd = sd(CO2))
ggplot(data=resp, aes(x = time, y = CO2_mean, group = amend, color = amend)) + geom_point() + geom_line() + theme_classic() + 
  xlab("Time (weeks)") + ylab("CO2 (ppm)") + geom_errorbar(aes(ymin=CO2_mean-CO2_sd, ymax=CO2_mean+CO2_sd), width=.2, position=position_dodge(0.1))
ggsave("RNA_virus_respiration.pdf", width=8)

# Statistical testing
resp <- read.delim("respiration.tsv", sep="\t", header=T) %>% 
  mutate(CO2_ppm = as.numeric(gsub(",", "\\.", CO2_ppm))) %>%
  mutate(time = case_when(
  sample_ID %like% "std" ~ NA,
  sample_ID %like% "blank" ~ 0,
  .default = as.numeric(gsub("W.*", "", gsub("^.*T", "", sample_ID)))
)) %>%
  mutate(amend = case_when(
    sample_ID %like% "P" ~ "P",
    .default = "No P"
  )) %>% 
  select(time, CO2_ppm, amend) %>% pivot_longer(cols = CO2_ppm, values_to = "CO2") %>%
  filter(!is.na(time))
resp_P <- resp %>% filter(time==0 | amend=="P")
resp_noP <- resp %>% filter(amend=="No P")
pairwise.wilcox.test(resp_P$CO2, resp_P$time) # 0 is significanly different from all time points p<0.05
pairwise.wilcox.test(resp_noP$CO2, resp_noP$time) # 0 is significanly different from all time points p<0.05
```
```{r}
# Simulate infection by RNA phages during week 1
mapping_stats <- read.delim("SBB/table_S1_viralRNA.csv", header=T, sep=";")
# This is where I need the levi and durna tables
levi <- levi %>% mutate(pct = pct/100)
durna <- durna %>% mutate(pct = pct/100)
mapping_stats <- mapping_stats %>%
  mutate(variable = paste(gsub("-", "", Time_Point), ".R.", gsub("_", ".", Sample_Name), sep="")) %>%
  mutate(pct_reads_mapped_to_viruses = pct_reads_mapped_to_viruses/100)
levi <- levi %>% left_join(mapping_stats, by="variable") 
durna <- durna %>% left_join(mapping_stats, by="variable") 
# RNA was extracted from 0.5 g soil, so I'm multiplying the cell number by 2 to normalize per g
levi <- levi %>% rename(levi_pct = pct) %>%
  mutate(levi_pct_reads = levi_pct * pct_reads_mapped_to_viruses) %>%
  mutate(levi_RNA_ng = levi_pct_reads * 2 * Total.metaT.RNA..ng.) %>%
  mutate(Time_Point = gsub("T-", "", Time_Point))
durna <- durna %>% rename(durna_pct = pct) %>%
  mutate(durna_pct_reads = durna_pct * pct_reads_mapped_to_viruses) %>%
  mutate(durna_RNA_ng = durna_pct_reads * 2 * Total.metaT.RNA..ng.) %>%
  mutate(Time_Point = gsub("T-", "", Time_Point))

levi_P <- levi %>% filter(Time_Point=="0" | str_detect(variable, "P"))
levi_noP <- levi %>% filter(str_detect(variable, "P", negate = TRUE))
durna_P <- durna %>% filter(Time_Point=="0" | str_detect(variable, "P"))
durna_noP <- durna %>% filter(str_detect(variable, "P", negate = TRUE))
# Panels A and B
levi_P_fig <- ggplot(levi_P, aes(x = Time_Point, y = levi_RNA_ng)) + geom_boxplot(fill = "purple") + theme_classic() + 
  xlab("Time (weeks)") + ylab("RNA from Leviviricetes (ng / g soil)") + ggtitle("With P") + ylim(0,7)
levi_noP_fig <- ggplot(levi_noP, aes(x = Time_Point, y = levi_RNA_ng)) + geom_boxplot(fill = "salmon") + theme_classic() +
  xlab("Time (weeks)") + ylab("RNA from Leviviricetes (ng / g soil)") + ggtitle("Without P") + ylim(0,7)
durna_P_fig <- ggplot(durna_P, aes(x = Time_Point, y = durna_RNA_ng)) + geom_boxplot(fill = "purple") + theme_classic() + 
  xlab("Time (weeks)") + ylab("RNA from Durnavirales (ng / g soil)") + ggtitle("With P") + ylim(0,2)
durna_noP_fig <- ggplot(durna_noP, aes(x = Time_Point, y = durna_RNA_ng)) + geom_boxplot(fill = "salmon") + 
  theme_classic() + xlab("Time (weeks)") + ylab("RNA from Durnavirales (ng / g soil)") + ggtitle("Without P") +
  ylim(0,2)
# ggarrange(levi_P_fig, levi_noP_fig, nrow = 2, ncol = 1)
# Statistical testing of RNA amount by time
kruskal.test(levi_RNA_ng ~ Time_Point, levi_P) # Kruskal-Wallis chi-squared = 6.9697, df = 3, p-value = 0.07287
pairwise.wilcox.test(levi_P$levi_RNA_ng, levi_P$Time_Point, p.adjust.method = 'BH')
#   0     1     2    
# 1 0.071 -     -    
# 2 0.071 0.937 -    
# 3 0.095 0.937 0.937
kruskal.test(levi_RNA_ng ~ Time_Point, levi_noP) # Kruskal-Wallis chi-squared = 5.29, df = 3, p-value = 0.1517
kruskal.test(durna_RNA_ng ~ Time_Point, durna_P) # Kruskal-Wallis chi-squared = 2.9048, df = 3, p-value = 0.4065
kruskal.test(durna_RNA_ng ~ Time_Point, durna_noP) # Kruskal-Wallis chi-squared = 1.697, df = 3, p-value = 0.6376

levi_P_mean <- levi_P %>% select(Time_Point, levi_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(levi_RNA_ng_mean = round(mean(levi_RNA_ng), 2), levi_RNA_ng_sd  = round(sd(levi_RNA_ng), 2)) %>%
  mutate(levi_RNA_ng_diff = levi_RNA_ng_mean - levi_RNA_ng_mean[1])
levi_noP_mean <- levi_noP %>% select(Time_Point, levi_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(levi_RNA_ng_mean = round(mean(levi_RNA_ng), 2), levi_RNA_ng_sd  = round(sd(levi_RNA_ng), 2)) %>%
  mutate(levi_RNA_ng_diff = levi_RNA_ng_mean - levi_RNA_ng_mean[1])
durna_P_mean <- durna_P %>% select(Time_Point, durna_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(durna_RNA_ng_mean = round(mean(durna_RNA_ng), 5), durna_RNA_ng_sd  = round(sd(durna_RNA_ng), 2)) 
durna_noP_mean <- durna_noP %>% select(Time_Point, durna_RNA_ng) %>%
  group_by(Time_Point) %>% 
  summarise(durna_RNA_ng_mean = round(mean(durna_RNA_ng), 5), durna_RNA_ng_sd  = round(sd(durna_RNA_ng), 2)) 
# The average Leviviricetes genome size is 4030 bp, weighing on average 1.31*10^-12 ng
# We can back-calculate how many phages the RNA from Leviviricetes represents 
# By picking a burst size we can calculate how many cells lysed on average
# RNA was extracted from 0.5 g soil, so I'm multiplying the cell number by 2 to normalize per g
levi_P_mean <- levi_P_mean %>% 
  mutate(num_virions = levi_RNA_ng_diff*2/(1.31e-12)) %>%
  #mutate(num_cells_BS_2 = num_virions/2) %>%
  #mutate(num_cells_BS_20 = num_virions/20) %>%
  mutate(num_cells_BS_200 = num_virions/200) %>%
  mutate(num_cells_BS_2000 = num_virions/2000) %>%
  mutate(num_cells_BS_20000 = num_virions/20000) %>%
  mutate(num_cells_BS_200000 = num_virions/200000) %>%
  pivot_longer(cols = starts_with("num_cells"), names_to = "burst_size", values_to = "num_cells") %>%
  mutate(burst_size =  gsub(".*_", "", burst_size))
levi_noP_mean <- levi_noP_mean %>% 
  mutate(num_virions = levi_RNA_ng_diff*2/(1.31e-12)) %>%
  #mutate(num_cells_BS_2 = num_virions/2) %>%
  #mutate(num_cells_BS_20 = num_virions/20) %>%
  mutate(num_cells_BS_200 = num_virions/200) %>%
  mutate(num_cells_BS_2000 = num_virions/2000) %>%
  mutate(num_cells_BS_20000 = num_virions/20000) %>%
  mutate(num_cells_BS_200000 = num_virions/200000) %>%
  pivot_longer(cols = starts_with("num_cells"), names_to = "burst_size", values_to = "num_cells") %>%
  mutate(burst_size =  gsub(".*_", "", burst_size))
# durnavirales are double stranded, so the average genome size has to be doubled
durna_P_mean <- durna_P_mean %>% 
  mutate(num_virions = durna_RNA_ng_mean*2/(2*3.68e-11)) %>%
  mutate(num_cells_BS_200 = num_virions/200) %>%
  mutate(num_cells_BS_2000 = num_virions/2000) %>%
  mutate(num_cells_BS_20000 = num_virions/20000) %>%
  mutate(num_cells_BS_200000 = num_virions/200000) %>%
  pivot_longer(cols = starts_with("num_cells"), names_to = "burst_size", values_to = "num_cells") %>%
  mutate(burst_size =  gsub(".*_", "", burst_size))
durna_noP_mean <- durna_noP_mean %>% 
  mutate(num_virions = durna_RNA_ng_mean*2/(2*3.68e-11)) %>%
  mutate(num_cells_BS_200 = num_virions/200) %>%
  mutate(num_cells_BS_2000 = num_virions/2000) %>%
  mutate(num_cells_BS_20000 = num_virions/20000) %>%
  mutate(num_cells_BS_200000 = num_virions/200000) %>%
  pivot_longer(cols = starts_with("num_cells"), names_to = "burst_size", values_to = "num_cells") %>%
  mutate(burst_size =  gsub(".*_", "", burst_size))
# Panels C and D
levi_P_mean_fig <- ggplot(data = levi_P_mean, aes(x = Time_Point, y = num_cells+1, colour = burst_size, group = burst_size)) +
  theme_classic() + xlab("Time (weeks)") + ylab("Mean total virocells / g soil") +
  scale_color_brewer(palette = "Purples", direction = -1) + geom_point() + geom_line() + 
  ggtitle("With P") + scale_y_continuous(trans='log10')
levi_noP_mean_fig <- ggplot(data = levi_noP_mean, aes(x = Time_Point, y = num_cells+1, colour = burst_size, group = burst_size)) +
  theme_classic() + xlab("Time (weeks)") + ylab("Mean total virocells / g soil") +
  scale_color_brewer(palette = "Reds", direction = -1) + geom_point() + geom_line() + 
  ggtitle("Without P")  + scale_y_continuous(trans='log10')
durna_P_mean_fig <- ggplot(data = durna_P_mean, aes(x = Time_Point, y = num_cells+1, colour = burst_size, group = burst_size)) +
  theme_classic() + xlab("Time (weeks)") + ylab("Mean total virocells / g soil") +
  scale_color_brewer(palette = "Purples", direction = -1) + geom_point() + geom_line() + 
  ggtitle("With P") + scale_y_continuous(trans='log10')
durna_noP_mean_fig <- ggplot(data = durna_noP_mean, aes(x = Time_Point, y = num_cells+1, colour = burst_size, group = burst_size)) +
  theme_classic() + xlab("Time (weeks)") + ylab("Mean total virocells / g soil") +
  scale_color_brewer(palette = "Reds", direction = -1) + geom_point() + geom_line() + 
  ggtitle("Without P") + scale_y_continuous(trans='log10')
ggarrange(levi_P_fig, levi_noP_fig, levi_P_mean_fig, levi_noP_mean_fig, nrow = 2, ncol = 2)
ggsave("fig5_lysis_simulation.pdf", height=6, width=8)
ggarrange(durna_P_fig, durna_noP_fig, durna_P_mean_fig, durna_noP_mean_fig, nrow = 2, ncol = 2)
ggsave("fig5_lysis_simulation_durna.pdf", height=6, width=8)
```
```{r}
#Separate PCoA of phages (Leviviricetes and Durnavirales) and other viruses
phage_comm <- wetup_tax %>% filter(Class %like% "Levivi" | Order %like% "Durnavi")
other_comm <- wetup_tax %>% filter(!(Class %like% "Levivi" | Order %like% "Durnavi"))

temp <- phage_comm %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = FALSE))
wetup_sqrt <- sqrt(temp[rowSums(temp) > 0,])
wetup_dist <- vegdist(t(wetup_sqrt), method="bray", na.rm = T)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_")) %>%
  mutate(ellps = as.factor(gsub("T[1-3]_", "", cond)))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample, group=ellps)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  theme_bw()
ggsave("RNA_phages_PCoA_wetup_sqrt.pdf", width=10)

temp <- other_comm %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = FALSE))
wetup_sqrt <- sqrt(temp[rowSums(temp) > 0,])
wetup_dist <- vegdist(t(wetup_sqrt), method="bray", na.rm = T)

pcoa <- pcoa(wetup_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame
pcoa_df$grp <- gsub("18O.*", "18O", gsub("\\.R.*", "", gsub("\\..*18O", "_18O", pcoa_df$sample)))
pcoa_df$grp[pcoa_df$grp %like% "T0"] <- "T0_16O"
pcoa_df$grp[pcoa_df$grp %like% "H2O"] <- str_c(substr(pcoa_df$grp[pcoa_df$grp %like% "H2O"], 1, 2), "16O", sep = "_")
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_")) %>%
  mutate(ellps = as.factor(gsub("T[1-3]_", "", cond)))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample, group=ellps)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  theme_bw()
ggsave("RNA_nonphages_PCoA_wetup_sqrt.pdf", width=10, height = 7)
```

```{r}
# ITS taxonomy
its_otus <- read.delim("qiime2/rep-seqs_ITS2_SE_vs_NCBI_ITS_besthit.tsv", sep="\t", header=F)
colnames(its_otus) <- c("asv", "acc", "pid", "qcovs", "qstart", "qend", "sstart", "send", "length", "evalue", "bitscore") 
its_otus$acc <- gsub("\\|", "", gsub("[a-z]*\\|", "", its_otus$acc))
its_otus$acc <- gsub("\\..*", "", its_otus$acc)
its_otus$ncbi_taxa_id <- accessionToTaxa(its_otus$acc, 'accessionTaxa.sql', version = 'base')
its_otus$ncbi_tax <- getTaxonomy(its_otus$ncbi_taxa_id, 'accessionTaxa.sql')
its_otus <- its_otus %>% select("asv", "ncbi_taxa_id", starts_with("ncbi_tax"))
its_otus_for_table <- data.frame(its_otus$ncbi_tax)
its_otus_for_table <- data.frame(its_otus$asv, its_otus_for_table)
colnames(its_otus_for_table)[1] <- "asv"
write_delim(its_otus_for_table, "ITS2_NCBI_taxonomy.tsv", delim = "\t")

# Add ITS ASV table
euks <- read.delim("qiime2/ITS2_SE_ASV_table.tsv", header = T, sep = "\t")
colnames(euks)[1:3] <- c("asv", "T.0.Dry.Soil.A", "T.0.Dry.Soil.B")
euks <- euks %>% left_join(its_otus_for_table, by = "asv") %>% filter(!(is.na(phylum))) %>% filter(phylum %like% "mycota")
euk_melt <- euks %>% column_to_rownames("asv") %>% pivot_longer(cols = starts_with("T."), names_to = "sample") %>% mutate(time = substr(sample, 3, 3)) %>% mutate(treat = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
))
euks_for_dist <- euks %>% column_to_rownames("asv") %>% select(starts_with("T."))
colnames(euks_for_dist) <- gsub("T\\.", "T", colnames(euks_for_dist))

# Plot
ggplot(euk_melt, aes(x = sample, y = value, fill = phylum)) + geom_bar(position = "fill", stat = "identity", na.rm = T) + theme_bw() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_brewer(palette = "Paired") + xlab("Sample") + ylab("Proportion")
ggsave("Fungi.pdf")
temp <- euk_melt %>% filter(phylum %like% "Ascomycota") %>% filter(!(is.na(class)))
ggplot(temp, aes(x = sample, y = value, fill = class)) + geom_bar(position = "fill", stat = "identity", na.rm = T) + theme_bw() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_brewer(palette = "Paired") + xlab("Sample") + ylab("Proportion")
ggsave("Ascomycota.pdf")

# Statistics
euk_dist <- vegdist(t(euks_for_dist), method = "bray")
design_comm <- read.delim("design_comm.txt", sep = "\t", header = T)
adonis2(euk_dist~time*Phos, data=design_comm)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = euk_dist ~ time * Phos, data = design_comm)
#           Df SumOfSqs      R2      F Pr(>F)  
# time       1   0.2687 0.06689 2.0253  0.014 *
# Phos       1   0.1391 0.03463 1.0486  0.351  
# time:Phos  1   0.1597 0.03974 1.2033  0.236  
# Residual  26   3.4500 0.85873                
# Total     29   4.0176 1.00000                
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

euks_sqrt <- sqrt(euks_for_dist)
euks_dist <- vegdist(t(euks_sqrt), method="bray", na.rm = T)
euks_cols <- factor(design_comm$cosm)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(euks_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = case_when(
    str_detect(sample, "T0") ~ "T0_16O",
    str_detect(sample, "T1.*18O") ~ "T1_18O",
    str_detect(sample, "T2.*18O") ~ "T2_18O",
    str_detect(sample, "T3.*18O") ~ "T3_18O",
    str_detect(sample, "T1.*H2O") ~ "T1_16O",
    str_detect(sample, "T2.*H2O") ~ "T2_16O",
    str_detect(sample, "T3.*H2O") ~ "T3_16O"
  )) 
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 1) +
  theme_bw()
ggsave("RNA_viruses_PCoA_ITS2_sqrt.pdf", width=10, height = 7)

# Statistics of fungal viruses community dynamics
myc <- wetup_tax %>% filter(taxonomy %like% "Narna") %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = F))
myc_sqrt <- sqrt(phg)
myc_dist <- vegdist(t(myc), method = "bray")
adonis2(myc_dist~time*Phos, data = design)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = phg_dist ~ time * Phos, data = design)
#           Df SumOfSqs      R2       F Pr(>F)    
# time       1   0.4353 0.07639  4.2968  0.001 ***
# Phos       1   1.4804 0.25979 14.6130  0.001 ***
# time:Phos  1   0.2371 0.04161  2.3406  0.014 *  
# Residual  35   3.5458 0.62222                   
# Total     38   5.6987 1.00000                   
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

phg_cols <- factor(design_comm$cosm)
phg_dist <- vegdist(t(phg_sqrt), method = "bray")

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(phg_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = case_when(
    str_detect(sample, "T0") ~ "T0_16O",
    str_detect(sample, "T1.*18O") ~ "T1_18O",
    str_detect(sample, "T2.*18O") ~ "T2_18O",
    str_detect(sample, "T3.*18O") ~ "T3_18O",
    str_detect(sample, "T1.*H2O") ~ "T1_16O",
    str_detect(sample, "T2.*H2O") ~ "T2_16O",
    str_detect(sample, "T3.*H2O") ~ "T3_16O"
  )) 
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 0.8) +
  theme_bw()
ggsave("RNA_viruses_PCoA_phages_sqrt.pdf", width=10, height = 7)
```
```{r}
# Dynamics of phages with iPHoP hosts
phages <- read.csv("iphop_genome_predictions_new.csv", header=T) %>% select(!c(Host.genome, Main.method)) %>% unique %>% group_by(Virus) %>% top_n(1, Confidence.score) %>% mutate(Host.taxonomy = gsub(";s.*", "", Host.taxonomy)) %>% unique() %>% rename(Contig = Virus)
phages_dyn <- wetup_tax %>% filter(Contig %in% phages$Contig) %>% mutate(virname = str_c("vOTU", row_number(), sep = "_")) %>% left_join(phages, by = "Contig")
data_cols  <- which(colnames(virs) %like% "^T[0-3]")
phages_dyn_rel <- phages_dyn
temp2 <- phages_dyn_rel[, data_cols] * 100 / colSums(wetup_tax[, data_cols])
phages_dyn_rel[, data_cols] <- temp2
phages_dyn_rel_melt <- phages_dyn_rel %>% pivot_longer(cols = matches("T[0-3]"), names_to = "sample") %>% mutate(time = substr(sample, 2, 2)) %>% mutate(treat = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
)) %>% mutate(host_gn = gsub("^.*g__", "", Host.taxonomy))

temp <- phages_dyn_rel
temp %>% mutate(Host.taxonomy = gsub(";o.*", "", Host.taxonomy)) %>% group_by(Host.taxonomy) %>% tally()
host_gn <- gsub("^.*g__", "", phages_dyn_rel$Host.taxonomy)

# Putative host dynamics
bacs <- read.delim("qiime2/phyloseq/otu_table_16S.tsv") %>% select(!NEGATIVE)
colnames(bacs)[1] <- "Feature.ID"
bacs_tax <- read.delim("qiime2/taxonomy_16S.tsv") %>% select(!Confidence)
bacs <- bacs %>% left_join(bacs_tax, by = "Feature.ID")
bacs_rlvt <- bacs %>% mutate(genus = gsub(";.*", "", gsub("^.*g__", "", Taxon))) %>% filter(genus %in% host_gn) # only 10  ASVs  have a matching host genus
colnames(bacs_rlvt)[2:3] <- c("T.0.Dry.Soil.A", "T.0.Dry.Soil.B")

host_gn_uniq <- unique(bacs_rlvt$genus)
data_cols  <- which(colnames(bacs_rlvt) %like% "^T\\.")
tot_count <- colSums(bacs[, data_cols])
bacs_rlvt_rel <- bacs_rlvt
bacs_rlvt_rel[ ,data_cols] <- bacs_rlvt[ ,data_cols]*100 / tot_count
bacs_rlvt_rel <- bacs_rlvt_rel %>% pivot_longer(cols = starts_with("T."), names_to = "sample") %>% mutate(time = substr(sample,3,3)) %>% mutate(treat = case_when(
  str_detect(sample, "P") ~ "P",
  .default = "noP"
))

for (gn in host_gn_uniq) {
  temp <- bacs_rlvt_rel %>% filter(genus == gn) %>% mutate(grp = str_c(Feature.ID, treat, sep = "_")) %>% group_by(time, treat) %>% summarise(tot = sum(value))
  my_pal = c(brewer.pal(12, "Paired"), "grey", "black")
  ggplot(data = temp, aes(x = time, y = tot, color = treat, group = treat)) + geom_point() + theme_bw() + geom_line(stat = "identity") + 
    scale_color_manual(values = my_pal)
  ggsave(str_c(gn, ".pdf"))
}

# Statistics
bac_for_dist <- bacs %>% column_to_rownames("Feature.ID") %>% rename(T.0.Dry.Soil.A = X1.Dry.Soil, T.0.Dry.Soil.B = X2.Dry.Soil) %>% select(starts_with("T."))
colnames(bac_for_dist) <- gsub("T\\.", "T", colnames(bac_for_dist))
bac_dist <- vegdist(t(bac_for_dist), method = "bray")
design_comm <- read.delim("design_comm.txt", sep = "\t", header = T)
adonis2(bac_dist~time*Phos, data=design_comm)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = bac_dist ~ time * Phos, data = design_comm)
#           Df SumOfSqs      R2      F Pr(>F)
# time       1   0.3989 0.03291 0.9558  0.671
# Phos       1   0.4528 0.03735 1.0848  0.161
# time:Phos  1   0.4177 0.03446 1.0008  0.459
# Residual  26  10.8522 0.89528              
# Total     29  12.1216 1.00000

bacs_sqrt <- sqrt(bac_for_dist)
bacs_dist <- vegdist(t(bacs_sqrt), method="bray", na.rm = T)
bacs_cols <- factor(design_comm$cosm)

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(bacs_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = case_when(
    str_detect(sample, "T0") ~ "T0_16O",
    str_detect(sample, "T1.*18O") ~ "T1_18O",
    str_detect(sample, "T2.*18O") ~ "T2_18O",
    str_detect(sample, "T3.*18O") ~ "T3_18O",
    str_detect(sample, "T1.*H2O") ~ "T1_16O",
    str_detect(sample, "T2.*H2O") ~ "T2_16O",
    str_detect(sample, "T3.*H2O") ~ "T3_16O"
  )) 
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 0.6) +
  theme_bw()
ggsave("RNA_viruses_PCoA_16S_sqrt.pdf", width=10, height = 7)

# Statistics of phage community dynamics
phg <- wetup_tax %>% filter(taxonomy %like% "Levivi" | taxonomy %like% "Durna") %>% column_to_rownames("Contig") %>% select(starts_with("T", ignore.case = F))
phg_sqrt <- sqrt(phg)
phg_dist <- vegdist(t(phg), method = "bray")
adonis2(phg_dist~time*Phos, data = design)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 999
# 
# adonis2(formula = phg_dist ~ time * Phos, data = design)
#           Df SumOfSqs      R2       F Pr(>F)    
# time       1   0.4353 0.07639  4.2968  0.001 ***
# Phos       1   1.4804 0.25979 14.6130  0.001 ***
# time:Phos  1   0.2371 0.04161  2.3406  0.014 *  
# Residual  35   3.5458 0.62222                   
# Total     38   5.6987 1.00000                   
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

phg_cols <- factor(design_comm$cosm)
phg_dist <- vegdist(t(phg_sqrt), method = "bray")

cols <- brewer.pal(11, "BrBG")[c(1:4,8:11)]

pcoa <- pcoa(phg_dist)
jac_variances <- data.frame(pcoa$values$Relative_eig) %>% 
  #select(PercVar = 'pcoa_bray.values.Relative_eig') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
pcoa_df <- data.frame(pcoa$vectors) %>% 
  rownames_to_column(var = "sample") %>% 
  data.frame %>%
  mutate(grp = case_when(
    str_detect(sample, "T0") ~ "T0_16O",
    str_detect(sample, "T1.*18O") ~ "T1_18O",
    str_detect(sample, "T2.*18O") ~ "T2_18O",
    str_detect(sample, "T3.*18O") ~ "T3_18O",
    str_detect(sample, "T1.*H2O") ~ "T1_16O",
    str_detect(sample, "T2.*H2O") ~ "T2_16O",
    str_detect(sample, "T3.*H2O") ~ "T3_16O"
  )) 
pcoa_df <- pcoa_df %>% mutate(shape_grp = ifelse(str_detect(sample, "P"), "P", "No_P")) %>%
  mutate(cond = paste(grp, shape_grp, sep="_"))
eigenvalues<-round(jac_variances[,2], digits = 4)*100
ggplot(pcoa_df, aes(x = Axis.1, y = Axis.2, label = sample)) +
  geom_point(aes(color=grp, shape=shape_grp), size = 4) +
  ylab(paste0('Co 2 ',eigenvalues[2],'%')) + #Extract y axis value from variance
  xlab(paste0('Co 1 ',eigenvalues[1],'%')) + #Extract x axis value from variance
  scale_color_manual(values=cols) +
  ggtitle('Square root transformed PCoA Ordination') +
  coord_fixed(ratio = 0.8) +
  theme_bw()
ggsave("RNA_viruses_PCoA_phages_sqrt.pdf", width=10, height = 7)
```


